---
title: "Mesocosm Spider Analyses March 27, 2020"
author: "Ana Miller-ter Kuile"
date: "3/30/2020"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_float:
      collapsed: no
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, include=FALSE, warning=FALSE, message=FALSE)
```

```{r load packages}
library(here) #tidy data
library(tidyverse) #tidy data
library(ggplot2) #visualize data
library(emmeans) #marginal means of models
library(rcompanion) #pairwise comparisions
library(effects) #dotplots and stuff
library(lattice) #again dotplots
library(DHARMa) #model diagnostics
library(MuMIn) #model diagnostics
library(glmmTMB) #mixed models
library(lme4) #mixed models
library(ggeffects) #ggpredict function
library(performance) #for binomail model fit
library(see) #for binomial model fit
library(coin) #wilcoxon test
library(MASS) #glm.nb
library(vegan) #rrarefy
library(cowplot) #plot grid at end
```

# Introduction

*This is part one of two of the analyses for this project. Part 2 will examine environmental contamination for field-collected predators.*

This code examines the diets of mesocosm predators (the spider *Heteropoda venatoria*), specifically looking for the detection and abundance of prey DNA in predators which we have fed a known diet item (*Oxya japonica*). 

Specifically, we are exploring:

#### Does surface sterilization alter prey ASV detection or read abundance by removing surface contamination?

```{r surface contamination, echo=FALSE, include=TRUE, out.width="100%", fig.align='center', fig.cap = "Figure 1: DNA contaminating the outside surfaces of predators may alter ecological inference of diet. This may be especially important in mesocosm environments where predators and prey may come into contact with each other or shared surfaces."}
knitr::include_graphics(here("Pictures", "surface_contam.png"))
```

## Why it matters

Mesocosms are regularly used in food web projects to examine the consumptive effects of predators on prey (i.e. Rudolf et al. 2014). DNA metabarcoding offers a new way to confirm the effects of predator consumption in these environments. However, there are important considerations to take into account in this environment. Specifically, we want to ensure that when we detect prey for a particular predator, that this prey DNA is prey that was ingested, rather than prey encountered by the predator in the mesocosm environment. More broadly, since mesocosms are controlled environments, they can also be used as a way to confirm the efficacy of DNA metabarcoding methods at detecting prey in predator diets. 

# Analyses outline

#### The data 

The data for these analyses consist of 19 spiders of the species *Heteropoda venatoria* collected on Palmyra Atoll in 2017. All spiders were starved in mesocosms for 12 hours and then fed a grasshopper (*Oxya japonica*). Spiders and grasshoppers were kept in mesocosms for 24 hours, after which all predators who had killed (and presumably ingested some or all of) the diet item were frozen at -20C. We surface sterilized 8 spider individuals by submerging them (after death in freezer) in 10% bleach for 2 minutes, and then washing them in deionized water for another 2 minutes. The remaining 11 spiders were not surface sterilized. Spiders were stored in 80% EtOH prior to DNA extraction. 

```{r experimental setup, echo=FALSE, include=TRUE, out.width="100%", fig.align='center', fig.cap = "Figure 2: The mesocosm surface contamination experimental setup with sample sizes."}
knitr::include_graphics(here("Pictures", "mesocosm_experiment.png"))
```

After DNA extraction, PCR amplification, and sequencing on an Illumina MiSeq platform, sequences were quality filtered and denoised using two common denoising algorithms (UNOISE3 and DADA2). Both of these denoising algorithms take into account DNA sequence abundance and quality to combine sequences into Amplicon Sequence Variants (ASVs). Unlike traditional assignment methods (such as operational taxonomic units (OTUs)), which cluster sequences based on 97% sequence similarity, ASV denoising takes into account both sequence quality and abundance, and can detect differences and separate sequences that differ by one or very few nucleotides (ASVs are sometimes refered to as exact sequence variant (ESVs)).

```{r ASV assignment figure, include=TRUE, echo=FALSE, out.width="100%", fig.align='center', fig.cap = "Figure 3: Denoising algorithms (including DADA2 and UNOISE3) consider sequence abundance and error rate in raw sequence datasets when removing sequences that have high error and low abundance and combining their reads with those of their most similar neighbor sequence with high abundance and low error (amplicon sequence variants or ASVs)."}
knitr::include_graphics(here("Pictures", "denoise_picture.png"))
```

Our final dataset for these mesocosm spiders includes ASVs assigned to known prey and the abundance of these ASVs in each sample for two denoising algorithms (UNOISE3 and DADA2). In addition, we have some mesocosm spiders (8) which we surface sterilized and some which we did not surface sterilize (11). *(Note: These numbers are different because of lower success in extracting DNA from surface sterilized mesocosm spiders.)*

#### The analyses

We will be looking at the whether surface sterilization alters our understanding of mesocosm feeding interactions in two ways:

1. Known prey detection per sample (*response: known prey ASV presence-absence*):

**Does surface sterilization alter our ability to detect known prey?**

  + Surface sterilization could increase the detection of known prey by removing surface contamination that may drown out actual prey during the PCR amplification and sequencing processes. 
  
  + Surface sterilization could decrease the detection of known prey by removing surface contamination coming from known prey encountered by the predator in the mesocosm environment either through shared surfaces or prey handling without ingestion.

 *glm(detection ~ surface sterilization x algorithm, family = binomial)*
 
2. Known prey abundance per sample (*response: known prey read abundance of rarefied reads*):

**Does surface sterilization alter how much known prey we can recover from the metabarcoding process?**
    
    A: Surface sterilization could increase the abundance of known prey, again by removing surface contamination that drowns out known diet. 
    
    B: Surface sterilization could decrease the abundance of known prey, again by removing known prey DNA on the surface of predators encountered in the mesocosm environment.

 *glm(abundance ~ surface sterilization x algorithm, family = poisson)*
  
#### The results of these analyses are summarised in the [summary table](#summary) along with some initial [ecological interpretations](#ecology) and [alternative hypothesis explorations](#alternative) .

#Set-up

## Import ASV tables

First, we will import the data from both denoising algorithms (DADA2 and UNOISE3) and the metadata attached to each of the spiders in our dataset, including whether they were surface sterilized or not.

```{r what the dfs are, eval=FALSE, include=TRUE}
d2_comm <- read.csv(here("data", "ASV_tables", "dada2_uc_asv_tab.tsv"), sep = "\t")
u3_comm <- read.csv(here("data", "ASV_tables", "unoise_uc_zotu_tab.txt"), sep = '\t')
 
#sample metadata 
metadata <- read.csv(here("data", "Sample_Metadata.csv")) 
```


```{r import datasets and change column names}
d2_comm <- read.csv(here("data", "ASV_tables", "dada2_uc_asv_tab.tsv"), sep = "\t")
u3_comm <- read.csv(here("data", "ASV_tables", "unoise_uc_zotu_tab.txt"), sep = '\t')

#rename X to ASV in all these tables.
d2_comm <- rename(d2_comm, "ASV" = "X")
u3_comm <- rename(u3_comm, "ASV" = "X.OTU.ID")

#rename all the sample names across dataframes for consistency
colnames(d2_comm) <- sapply(str_split(colnames(d2_comm), "_"), function(x){return(x[[1]])})
colnames(d2_comm) <- str_remove(colnames(d2_comm), "\\.")

d2_comm <- rename(d2_comm, "CL1" = "CL12")
d2_comm <- rename(d2_comm, "CL4" = "CL42")

colnames(u3_comm) <- sapply(str_split(colnames(u3_comm), "S"), function(x){return(x[[1]])})
u3_comm <- rename(u3_comm, "ASV" = "A")

#sample metadata
metadata <- read.csv(here("data", "Sample_Metadata.csv"))
```

The ASV tables are matrices with ASVs as rows, and samples as columns:

```{r sample df heads, include=TRUE}
d2_comm[1:10, 1:10]
```

And the metadata includes important data for each sample:

```{r metadata df head, include = TRUE}
str(metadata)
```

## Assign taxonomies

For each of the ASVs in each of these tables, we used BLAST (with GenBank) and the BOLD ID Engine to assign taxonomies to these ASVs. We will import the dataframes of these taxonomic assignments for each algorithm and then subset the taxonomies that belong to the known prey item (here, either *Acrididae* or *Oxya* depending on the taxonomic assignment tool).

```{r example of taxonomy DBs, eval=FALSE, include=TRUE}
#DADA2#
#import both NCBI and BOLD taxonomies
d2_ncbi <- read.csv(here("6_taxonomic_assignment", "MEGAN", "dada2_unclean",
                            "dada2_uc_ncbi_taxa.csv"))

d2_bold <- read.csv(here("6_taxonomic_assignment", "BOLD", "dada2_uc", 
                            "dada2_uc_taxa_bold.csv"))
 
#Have the same for UNOISE3 
#import both NCBI and BOLD taxonomies
u3_ncbi <- read.csv(here("6_taxonomic_assignment", "MEGAN", "unoise_unclean",
                            "unoise_uc_ncbi_taxa.csv"))
u3_bold <- read.csv(here("6_taxonomic_assignment", "BOLD", "usearch_uc", 
                            "unoise_uc_bold_taxa.csv"))
```


```{r assign taxonomies}
#DADA2#
#import both NCBI and BOLD taxonomies
d2_ncbi <- read.csv(here("6_taxonomic_assignment", "MEGAN", "dada2_unclean",
                            "dada2_uc_ncbi_taxa.csv"))
#this is everything that got a taxonomic assignment through MEGAN
d2_all <- read.csv(here("6_taxonomic_assignment", "MEGAN", "dada2_unclean",
                           "dada2_uc_all.csv"))
d2_bold <- read.csv(here("6_taxonomic_assignment", "BOLD", "dada2_uc", 
                            "dada2_uc_taxa_bold.csv"))
d2_bold <- rename(d2_bold, "ASV" = "Query.ID")

#Sort by "predator", "prey", and "non-diet" before merging into one 
#dataframe for assigning to the abundance table!
#this following ifelse set will depend on the taxonomy of the set of samples
#and for this one, these are the categories that seemed to be appropriate based
#on the taxonomy list after looking at it
#sort into categories for bold:
d2_bold$taxonomy <- ifelse(
  d2_bold$ID == "Heteropoda venatoria", "predator",
  ifelse(d2_bold$Kingdom == "Fungi", "non-diet", 
         ifelse(d2_bold$Phylum == "Arthropoda" & d2_bold$ID != "Heteropoda venatoria", 
                "prey", "non-diet"
         )))

#sort into categories for ncbi:
d2_ncbi$taxonomy <- ifelse(
  d2_ncbi$ID == "Sparassidae" | d2_ncbi$ID == "Heteropoda venatoria", "predator",
  ifelse(d2_ncbi$Phylum == "Arthropoda" & d2_ncbi$ID != "Sparassidae" & d2_ncbi$ID != "Heteropoda venatoria", 
         "prey", "non-diet"
  ))

#this is an anti-join to subset from ALL MEGAN hits for those that aren't within
#diet categories, but which still got assignments in MEGAN
d2_nond <- d2_all %>%
  anti_join(d2_ncbi, by = "ASV")
#this gives these all the category of "non-diet" to distinguish from those that didn't
#get any taxonomic assignment at all. 
d2_nond$taxonomy <- d2_nond$Category

#thus, the taxonomy variable has 4 levels: predator, prey, non-diet, non-hit (which
#we will assign below after merging with full ASV table)
#subset only the variables of interest from these two dataframes before merging
d2_bold_id <- d2_bold %>%
  dplyr::select(ASV, ID, Order, taxonomy)

d2_ncbi_id <- d2_ncbi %>%
  dplyr::select(ASV, ID, Order, taxonomy)

#join them together by ASV and taxonomy
d2_id_both <- d2_bold_id %>%  
  full_join(d2_ncbi_id, by = c("ASV", "taxonomy"))

#join these both again with the non-diet hits
d2_id_all <- d2_id_both %>%
  full_join(d2_nond, by = c("ASV", "taxonomy"))

d2_id_all <- d2_id_all %>%
  rename(ID_bold = ID.x,
         Order_bold = Order.x,
         ID_ncbi = ID.y,
         Order_ncbi = Order.y,
         ID_nondiet = ID,
         Order_nondiet = Order)

#this binds them all to the community matrix for analyses
d2_id_tab <- d2_comm %>%
  left_join(d2_id_all, by = "ASV")

#set NA taxonomies from join with "no hit" since they didn't match to any
#taxonomic assignments within our prey, predator, or non-prey categories
d2_id_tab$taxonomy <- replace_na(d2_id_tab$taxonomy, "no hit")

#write this to a file for importing into the field spider analyses
d2_taxonomies <- d2_id_tab %>%
  dplyr::select(ASV, ID_bold, Order_bold, ID_ncbi, Order_ncbi, ID_nondiet, Order_nondiet, taxonomy)
write.csv(d2_taxonomies, here("data", "d2_uc_tax_ass.csv"))

#UNOISE3##
#import both NCBI and BOLD taxonomies
u3_ncbi <- read.csv(here("6_taxonomic_assignment", "MEGAN", "unoise_unclean",
                            "unoise_uc_ncbi_taxa.csv"))
u3_bold <- read.csv(here("6_taxonomic_assignment", "BOLD", "usearch_uc", 
                            "unoise_uc_bold_taxa.csv"))
u3_bold <- rename(u3_bold, "ASV" = "Query.ID")

#Sort by "predator", "prey", and "non-diet" before merging into one 
#dataframe for assigning to the abundance table!
#this following ifelse set will depend on the taxonomy of the set of samples
#and for this one, these are the categories that seemed to be appropriate based
#on the taxonomy list after looking at it
#sort into categories for bold:
u3_bold$taxonomy <- ifelse(
  u3_bold$ID == "Heteropoda venatoria", "predator",
  ifelse(u3_bold$Kingdom == "Fungi" | u3_bold$Class == "Mammalia", "non-diet", 
         ifelse(u3_bold$Phylum == "Arthropoda" | u3_bold$Class == "Reptilia" & u3_bold$ID != "Heteropoda venatoria", 
                "prey", "non-diet"
         )))

#sort into categories for ncbi:
u3_ncbi$taxonomy <- u3_ncbi$Category

#thus, the taxonomy variable has 4 levels: predator, prey, non-diet, non-hit (which
#we will assign below after merging with full ASV table)
#subset only the variables of interest from these two dataframes before merging
u3_bold_id <- u3_bold %>%
  dplyr::select(ASV, ID, Order, taxonomy)

u3_ncbi_id <- u3_ncbi %>%
  dplyr::select(ASV, ID, Order, taxonomy)

#join them together by ASV and taxonomy
u3_id_all <- u3_bold_id %>%  
  full_join(u3_ncbi_id, by = c("ASV", "taxonomy"))

u3_id_all <- u3_id_all %>%
  rename(ID_bold = ID.x,
         Order_bold = Order.x,
         ID_ncbi = ID.y,
         Order_ncbi = Order.y)

#this binds them all to the community matrix for analyses
u3_id_tab <- u3_comm %>%
  left_join(u3_id_all, by = "ASV")

#set NA taxonomies from join with "no hit" since they didn't match to any
#taxonomic assignments within our prey, predator, or non-prey categories
u3_id_tab$taxonomy <- replace_na(u3_id_tab$taxonomy, "no hit")

#write to a file for import into field analyses
u3_taxonomies <- u3_id_tab %>%
  dplyr::select(ASV, ID_bold, Order_bold, ID_ncbi, Order_ncbi, taxonomy)
write.csv(u3_taxonomies, here("data", "u3_uc_tax_ass.csv"))
```

Each taxonomic assignment includes identifiers from the Kingdom level down to Order, Family, Genus, or Species. Here are a few examples of taxonomies assigned to order and lower:

```{r ex taxonomy assignment, include =TRUE, echo=FALSE}
d2_ncbi %>%
  dplyr::select(ASV, Order, Family, Genus, Species) %>%
  filter(ASV %in% c("ASV_1", "ASV_100", "ASV_105", "ASV_106", "ASV_154", 
                    "ASV_5", "ASV_6", "ASV_107"))
```

# Known prey ASVs: Analyses

We are first going to look at whether surface sterilization changes the detection of known prey ASVs in our dataset. Again, roughly half of the samples were surface sterilized and the other half were not surface sterilized. Furthermore, we ran two different bioinformatic algorithms on ALL samples. We will be running a generalized linear model with the full format:

glm(detection ~ surface_sterilization*algorithm, family = binomial)

And then we will do model selection altering just the surface sterilization term, since this is the term we are interested in in the model. This full model is asking whether surface sterilization or denoising algorithm matter, and whether the relationship with them varies depending on the algorithm considered. 

### Data set up: subset mesocosm spiders, rarefy samples, and subset known prey ASVs

Right now, this dataset includes all the spiders in our project, including those which we will analyze in Part 2 (field spiders) and all ASVs, including both our known prey ASVs and everything else detected in these samples, so we will need to subset the mesocosm spiders and the known prey ASVs from this dataset. After subsetting the mesocosm spiders and before subsetting the known prey ASVs and doing any analyses, however, we need to rarefy our samples to account for unequal sequencing depths that result from random sequencing preferences in the sequencing process on the Illumina MiSeq. To illustrate, if we consider the number of sequences in each sample in our UNOISE3 dataset, we can see that read abundance can vary a lot across samples:

```{r example sequence depth diffs setup}
lab <- as.character(metadata$sample[which(metadata$Source == "LAB")]) #vector of the sample names for lab samples

u3_commr <- u3_comm
rownames(u3_commr) <- u3_commr$ASV
u3_commr <- u3_commr %>%
  dplyr::select(all_of(lab))
```

```{r example seq depth diff, include=TRUE}
hist(colSums(u3_commr))

#max read abundance 
max(colSums(u3_commr))

#min read abundance
min(colSums(u3_commr))
```

This amount of variation in sequencing depth could lead to potentially misleading interpretations of any known prey detection or read abundance derived from this distribution. Thus, we will use the sample with the lowest total sequencing depth to rarefy our dataset so that we are comparing samples with similar sampling effort. 

```{r rarefying datsets}
#using u3_commr created above and:
d2_commr <- d2_comm
rownames(d2_commr) <- d2_commr$ASV
d2_commr <- d2_commr %>%
  dplyr::select(all_of(lab))
```

```{r rarefy, include=TRUE}
u3rare <- min(colSums(u3_commr))
u3rare #55205
d2rare <- min(colSums(d2_commr))
d2rare #50345

set.seed(1)
u3_comm_rare <- as.data.frame(t(rrarefy(t(u3_commr), sample = u3rare)))
d2_comm_rare <- as.data.frame(t(rrarefy(t(d2_commr), sample = d2rare)))
```

Now, all our samples have equal sampling depths, allowing for comparisons across samples:

```{r sampling depth example, include=TRUE}
colSums(u3_comm_rare)
```

### Subset just prey from this dataset

Once we've rarefied, we can subset only the prey DNA from this dataset, and start our analyses: 

```{r known prey read binding and selection, include = TRUE}
u3_comm_rare$ASV <- rownames(u3_comm_rare)
d2_comm_rare$ASV <- rownames(d2_comm_rare)

#this binds them all to the community matrix for analyses
u3_rare <- u3_comm_rare %>%
  left_join(u3_id_all, by = "ASV") %>% #combine with taxonomy DF
  filter(ID_ncbi == "Acrididae") %>% #select known diet only
  dplyr::select(-ID_bold, -ID_ncbi) %>% #remove taxonomy columns
  group_by(ASV) %>% 
  gather(sample, reads, HEV07:HEV29, factor_key = TRUE) %>% #make long while grouped by ASV
  group_by(sample) %>%
  summarise(reads_u3 = sum(reads)) #summarise total prey reads by sample 

#do same for Dada2

#this binds them all to the community matrix for analyses
d2_rare <- d2_comm_rare %>%
  left_join(d2_id_all, by = "ASV") %>% #combine with taxonomy DF 
  filter(ID_ncbi == "Acrididae") %>% #select known diet only
  dplyr::select(-ID_bold, -ID_ncbi, -ID_nondiet) %>% #remove taxonomy columns
  group_by(ASV) %>% 
  gather(sample, reads, HEV07:HEV29, factor_key = TRUE) %>% #make long while grouped by ASV
  group_by(sample) %>%
  summarise(reads_d2 = sum(reads)) #summarise total prey reads by sample 

all_rare <- u3_rare %>%
  left_join(d2_rare, by = "sample") %>%
  gather(pipeline, reads, reads_u3:reads_d2) %>%
  left_join(metadata, by = "sample")
```

Now we're ready to do both our presence-absence AND abundance analyses for these mesocosm spiders!

### Binomial GLM of prey detection

```{r set up for binomial glmm, include=TRUE}
#create variable for surface sterilization
all_rare$surf_ster <- as.factor(ifelse(all_rare$Sterilized == "NS", 0, 1)) 

#create variable for presence-absence 
all_rare$Presence <- ifelse(all_rare$reads == 0, 0, 1)
```

Now we can run a set of models assessing whether surface sterilization is important for these samples. Our models will all include algorithm, since we aren't interested in algorithm-specific effects. We will be altering the form of surface sterilization in the model, since we are interested in this effect on the presence of known prey. Specifically, the full model will include the interaction between surface sterilization and algorithm, which means that the effect of surface sterilization can vary by algorithm. The second model will not include the interaction term, which means that surface sterilization acts the same for both algorithms. Lastly, we will run a "null" model which will not include surface sterilization. 

```{r model for ASV presence, include=TRUE}
ASV_ster_mod <- glm(Presence ~ surf_ster*pipeline,
                    data = all_rare, 
                    family = binomial)

ASV_ster_mod2 <- glm(Presence ~ surf_ster + pipeline,
                     data = all_rare,
                     family = binomial)

ASV_ster_null <- glm(Presence ~ pipeline,
                     data = all_rare,
                     family = binomial)

AICc(ASV_ster_mod, ASV_ster_mod2, ASV_ster_null)
```

Based on these AIC values, the model with surface sterilization and no interaction term is the best fit for these data. 

And now we can look at some summaries of this model, which show that it's a good model, and that surface sterilization is a significant term in the model.

```{r model ASV summaries, include=TRUE}
summary(ASV_ster_mod2)
```

```{r model ASV fit, include =TRUE, fig.show='hide'}
simulationOutput_ASV <- simulateResiduals(fittedModel = ASV_ster_mod2) 
ASV_fit <- plot(simulationOutput_ASV, asFactor=TRUE)
```

This is a fairly good model fit.We can further look at all the effects in this model by plotting the effects and then comparing the pairwise estimated marginal means of surface sterilization vs. no surface sterilization for each pipeine.

```{r model ASV pairwise, include=TRUE, fig.show='hide'}
plot(allEffects(ASV_ster_mod2))
model.emm_ASV <- emmeans(ASV_ster_mod2, pairwise ~ surf_ster | pipeline)
pairs(model.emm_ASV$emmeans)
```

What these results show is that surface sterilization significantly decreases known prey detection for both denoising algorithms.  

```{r ASV visualization set up}
#let's visualize this then
present <- all_rare %>%
  group_by(pipeline, Sterilized) %>%
  filter(reads > 0) %>%
  summarise(present = n()) 

total <- all_rare %>%
  group_by(pipeline, Sterilized) %>%
  summarise(total = n()) %>%
  dplyr::select(pipeline, Sterilized, total)

ASVs_known <- present %>%
  left_join(total, by = c("pipeline" = "pipeline", "Sterilized" = "Sterilized"))

ASVs_known <- ASVs_known %>%
  mutate(presence = present/total)

ASVs_known1 <- ASVs_known %>%
  group_by(Sterilized) %>%
  summarise(mean = mean(presence), se = sd(presence)/sqrt(2))

```

```{r ASV detection visualization by sterilized, include=TRUE, echo=FALSE}
(a <- ggplot(ASVs_known1, aes(x = Sterilized, y = mean, fill = Sterilized)) +
  geom_bar(stat="identity", position= "dodge") +theme_bw() +
  labs(x = "Surface sterilization treatment", y = "Percent Known Prey ASV Detection",
       title = "Percent of samples with known prey detected") +
  scale_fill_manual(values = c("#7B8D65","#F29979")) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean +se), width = 0.2) +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized")) +
  theme(legend.position = "none"))
```

#### Summary: Sterilization significantly decreased the detection of prey ASVs for both denoising algorithms. 

# Known prey reads: Analyses

Now, we are going to look at whether surface sterilization changes the abundance of known prey reads in our dataset. Again, we will be running a generalized linear model, now with a full format:

glm(abundance ~ surface_sterilization*algorithm, family = poisson)

And then we will do model selection altering just the surface sterilization term, since this is the term we are interested in in the model. This full model is asking whether surface sterilization or denoising algorithm matter, and whether the relationship with them varies depending on the algorithm considered. 

### GLM for surface sterilization effects on read abundance

Now we can run a set of models assessing whether surface sterilization is important for read abundances in these samples. Our models will all include algorithm, since we aren't interested in algorithm-specific effects. We will be altering the form of surface sterilization in the model, since we are interested in this effect on the presence of known prey. Specifically, the full model will include the interaction between surface sterilization and algorithm, which means that the effect of surface sterilization can vary by algorithm. The second model will not include the interaction term, which means that surface sterilization acts the same for both algorithms. Lastly, we will run a "null" model which will not include surface sterilization. 

```{r sterilization and abundance models poisson, include=TRUE}
read_ster_mod <- glm(reads ~ surf_ster*pipeline,
                    data = all_rare, 
                    family = poisson)

read_ster_mod2 <- glm(reads ~ surf_ster + pipeline,
                     data = all_rare,
                     family = poisson)

read_ster_null <- glm(reads ~ pipeline,
                     data = all_rare,
                     family = poisson)

AICc(read_ster_mod, read_ster_mod2, read_ster_null) #first and second model really similar, but let's check model assumptions before we move on
```

```{r ster abundance model fits poisson, include=TRUE, fig.show = 'hide'}
simulationOutput_knownprey <- simulateResiduals(fittedModel = read_ster_mod2) 
knownpreyfit <- plot(simulationOutput_knownprey) #these look okay?
knownpreyzi <- testZeroInflation(simulationOutput_knownprey) #not zero inflated
knownpreyod <- testDispersion(simulationOutput_knownprey) 
#for both models, serious problems with model assumptions, so we will try nb.

```

These models are not great, so we will fit with a negative binomial model instead:

```{r ster and abundance models NB, include=TRUE}
read_ster_mod3 <- glm.nb(reads ~ surf_ster*pipeline,
                    data = all_rare)

read_ster_mod4 <- glm.nb(reads ~ surf_ster + pipeline,
                     data = all_rare)

read_ster_null2 <- glm.nb(reads ~ pipeline,
                     data = all_rare)

AICc(read_ster_mod3, read_ster_mod4, read_ster_null2) #now second is better fit, but let's see about those model assumptions
```

```{r model fits abund and NB, include =TRUE, fig.show = 'hide'}
simulationOutput_knownprey <- simulateResiduals(fittedModel = read_ster_mod4) 
knownpreyfit <- plot(simulationOutput_knownprey) #these look okay?
knownpreyzi <- testZeroInflation(simulationOutput_knownprey) #not zero inflated
knownpreyod <- testDispersion(simulationOutput_knownprey) 
#this is a lot better!
```

These look better.

```{r residuals NB ster model, include = TRUE, echo=FALSE, fig.show='hide'}
plot(residuals(read_ster_mod4)) 
```


```{r reads model summary and marginal means, include = TRUE, fig.show='hide'}
summary(read_ster_mod4)

plot(allEffects(read_ster_mod4))

model.emm_reads <- emmeans(read_ster_mod4, pairwise ~ surf_ster | pipeline)

pairs(model.emm_reads)
```

This model shows that sterilization significantly increases prey reads for both piplines. We can visualize the pattern here:

```{r visualize prey read abundance, echo=FALSE, include=TRUE}
#let's visualize this then
all_rare_sum <- all_rare %>%
  group_by(Sterilized) %>%
  summarise(mean = mean(reads), n(), se = sd(reads)/sqrt(`n()`))

ggplot(all_rare_sum, aes(x = Sterilized, y = mean, fill = Sterilized)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = mean-se, ymax=mean+se), width = 0.2) +
  labs(x = "Surface sterilization treatment", y = "Rarefied prey reads", title = "Number of known prey reads per sample") +
  theme_bw() + scale_fill_manual(values = c("#7B8D65","#F29979")) +
  scale_y_log10() + theme(legend.position = "none") +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized")) 
```

#### Summary Version 1: Sterilization significantly increased the amount of known prey in samples. 

### Caveat: an outlier driving pattern?

When looking at the distribution of read abundances in these samples, however, we noticed one thing, which is that one sample in the sterilized group has a read abundance an order of magnitude greater than any other sample:

```{r hist of read abundances, include=TRUE}
hist(all_rare$reads)
```

And when we remove that sample, all samples have more equal read abundances in them:

```{r hist of read abundances minus outlier, include=TRUE}
all_rare1 <- all_rare %>%
  filter(sample != "HEV07")

hist(all_rare1$reads)
```

Without this outlier, let's reassess our model fits:

```{r sterilization and abundance models poisson remove outlier, include=TRUE}
read_ster_mod_o <- glm(reads ~ surf_ster*pipeline,
                    data = all_rare1, 
                    family = poisson)

read_ster_mod_o2 <- glm(reads ~ surf_ster + pipeline,
                     data = all_rare1,
                     family = poisson)

read_ster_null_o <- glm(reads ~ pipeline,
                     data = all_rare1,
                     family = poisson)

AICc(read_ster_mod_o, read_ster_mod_o2, read_ster_null_o) #second model best, including model2 without the interaction, thought its within two AIC values of the model with the interaction
```

```{r model fits read abundance sans outlier, include=TRUE, fig.show = 'hide'}
simulationOutput_knownprey <- simulateResiduals(fittedModel = read_ster_mod_o2) 
knownpreyfit <- plot(simulationOutput_knownprey) #these look okay?
knownpreyzi <- testZeroInflation(simulationOutput_knownprey) #not zero inflated
knownpreyod <- testDispersion(simulationOutput_knownprey) 
#for both models, serious problems with model assumptions, so we will try nb.
```

These models don't meet assumptions, so we will fit some negative binomial models and see how these work out: 

```{r nb models of read abundance sans outlier, include=TRUE}
read_ster_mod_o3 <- glm.nb(reads ~ surf_ster*pipeline,
                    data = all_rare1)

read_ster_mod_o4 <- glm.nb(reads ~ surf_ster + pipeline,
                     data = all_rare1)

read_ster_null_o2 <- glm.nb(reads ~ pipeline,
                     data = all_rare1)

AICc(read_ster_mod_o3, read_ster_mod_o4, read_ster_null_o2) #4 and null are similar to each other

```

The model with surface sterilization (and no interaction) and the null model have equivalent AIC values. While this means we should go with the null model, I am going to assess the model with sterilization as well. 

```{r model fits of nb without outlier, include=TRUE, fig.show = 'hide'}
simulationOutput_knownprey <- simulateResiduals(fittedModel = read_ster_mod_o4) 
knownpreyfit <- plot(simulationOutput_knownprey) #these look okay?
knownpreyzi <- testZeroInflation(simulationOutput_knownprey) #not zero inflated
knownpreyod <- testDispersion(simulationOutput_knownprey)
plot(residuals(read_ster_mod_o4))
#for both models, serious problems with model assumptions, so we will try nb.
```

These all look pretty good, so let's look at the pattern: 

```{r summaries of nb reads model sans outlier, include=TRUE, fig.show='hide'}
summary(read_ster_mod_o4)

plot(allEffects(read_ster_mod_o4))

model.emm_reads <- emmeans(read_ster_mod_o4, pairwise ~ surf_ster | pipeline)

pairs(model.emm_reads$emmeans)
```

The between surface sterilization treatment patterns for both algorithms are marginally significant with surface sterilization reducing read abundance in samples.

```{r visualize read abundance pattern sans outlier, include=TRUE, echo=FALSE}
all_rare_sum1 <- all_rare1 %>%
  group_by(Sterilized) %>%
  summarise(mean = mean(reads), n(), se = sd(reads)/sqrt(`n()`)) %>%
  glimpse()

(b <- ggplot(all_rare_sum1, aes(x = Sterilized, y = mean, fill = Sterilized)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = mean-se, ymax=mean+se), width = 0.2) +
  labs(x = "Surface sterilization treatment", y = "Rarefied prey reads", title = "Number of known prey reads per sample") +
  theme_bw() + scale_fill_manual(values = c("#7B8D65","#F29979")) +
  scale_y_log10() + theme(legend.position = "none") +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized"))) 
```

#### Summary Version 2: Surface sterilization reduces known prey read abundance marginally. 

# Overall summary {#summary}

```{r summary table, include = TRUE, echo=FALSE}
headers <- c("", "Outcome")
row1 <- c("Known prey ASV detection", "Surface sterilization decreases detection of known prey")
row2 <- c("Known prey read abundance", "Surface sterilization decreases* known prey read abundance")
row3 <- c("*after removing an outlier", "")

summary_table <- as.data.frame(rbind(headers, row1, row2, row3))
colnames(summary_table) <- summary_table$headers
knitr::kable(summary_table[2:4,], col.names = headers, row.names=FALSE)


```

```{r presence value}
presence_value <- ASVs_known %>%
  filter(Sterilized == "SS", pipeline == "reads_d2") %>%
  ungroup() %>%
  dplyr::select(present)

presence_value = presence_value$present
```
  
  
# Biological/Ecological significance: {#ecology}

Hypothesis A: **Known prey is surface contamination from shared environment.** In a mesocosm environment, predators and prey may come into contact with each other in the mesocosm environment, either through shared surfaces or through the process of prey capture and handling. In our DNA metabarcoding data, some of the DNA that could be assigned as "ingested prey" may be contamination from shared space and surface contact in the mesocosm. While our mesocosms were simplified environments in which we could observe one predator individual interacting with one prey item, other mesocosm experiments can include multiple individuals or species of predators and prey that may or may not be large enough to observe interacting. In these contexts, confirming that prey detection is due to ingestion versus environmental contact is crucial for biological inference. The lack of detection in some surface sterilized individuals (5 or `r 5/8` of total) is also an interesting observation in our study, and may highlight that just because prey handling and mortality is observed, prey ingestion may not have occured. Alternatively, lack of detection represents limitations to the DNA metabarcoding approach to assigning prey items in predator diets. 

```{r summary graphs again, include=TRUE, out.width = '100%', echo=FALSE}
plot_grid(a, b)
```

# Alternative significance: {#alternative}

Hypothesis B: **The lack of detection could be deterioration/degradation of DNA from the surface sterilization process.**

A few lines of evidence we can use to explore this:

####1. Amount of predator DNA in each sample (this should be equal if sterilization *does not* deteriorate/degrade DNA).

Predator DNA read abundance does not decrease with surface sterilization:

```{r predator DNA, include = TRUE, fig.show = 'hide'}
#this binds them all to the community matrix for analyses
u3_pred <- u3_comm_rare %>%
  left_join(u3_id_all, by = "ASV") %>% #combine with taxonomy DF
  filter(taxonomy == "predator") %>% #select predator DNA only
  dplyr::select(-ID_bold, -ID_ncbi) %>% #remove taxonomy columns
  group_by(ASV) %>% 
  gather(sample, reads, HEV07:HEV29, factor_key = TRUE) %>% #make long while grouped by ASV
  group_by(sample) %>%
  summarise(reads_u3 = sum(reads)) %>% #summarise total prey reads by sample 
  left_join(metadata, by = "sample")

#do same for Dada2

#this binds them all to the community matrix for analyses
d2_pred <- d2_comm_rare %>%
  left_join(d2_id_all, by = "ASV") %>% #combine with taxonomy DF 
  filter(taxonomy == "predator") %>% #select predator DNA only
  dplyr::select(-ID_bold, -ID_ncbi, -ID_nondiet) %>% #remove taxonomy columns
  group_by(ASV) %>% 
  gather(sample, reads, HEV07:HEV29, factor_key = TRUE) %>% #make long while grouped by ASV
  group_by(sample) %>%
  summarise(reads_d2 = sum(reads))  %>% #summarise total prey reads by sample 
  left_join(metadata, by = "sample")

#all_pred <- u3_pred %>%
#  left_join(d2_pred, by = "sample") %>%
#  gather(pipeline, reads, reads_u3:reads_d2) %>%
#  left_join(metadata, by = "sample")

#ggplot(all_pred, aes(x = Sterilized, y = reads, fill = pipeline)) +
#  geom_boxplot()

#all_pred$surf_ster <- ifelse(all_pred$Sterilized == "NS", 0, 1)
library(glmmTMB)

#all_pred <- all_pred %>%
#  filter(!sample == "HEV07")

#hist(u3_pred$reads_u3)
#hist(d2_pred$reads_d2)
#U3:
u3_pred <- u3_pred %>%
  filter(!sample == "HEV07")

u3_pred$surf_ster <- ifelse(u3_pred$Sterilized == "NS", 0, 1)
read_pred_u3 <- glmmTMB(reads_u3 ~ surf_ster,
                    data = u3_pred,
                 family = truncated_genpois)

read_pred_u3_null <- glmmTMB(reads_u3 ~ 1,
                    data = u3_pred,
                    family = truncated_genpois)

AICc(read_pred_u3,read_pred_u3_null) 

simulationOutput_pred <- simulateResiduals(fittedModel = read_pred_u3_null) 
predfit <- plot(simulationOutput_pred) 
predzi <- testZeroInflation(simulationOutput_pred) 
predod <- testDispersion(simulationOutput_pred)
plot(residuals(read_pred_u3))

#DADA2
d2_pred <- d2_pred %>%
  filter(!sample == "HEV07")

d2_pred$surf_ster <- ifelse(d2_pred$Sterilized == "NS", 0, 1)

read_pred_d2 <- glmmTMB(reads_d2 ~ surf_ster,
                    data = d2_pred,
                 family = truncated_genpois)

read_pred_d2_null <- glmmTMB(reads_d2 ~ 1,
                    data = d2_pred,
                    family = truncated_genpois)

AICc(read_pred_d2,read_pred_d2_null) 

simulationOutput_pred <- simulateResiduals(fittedModel = read_pred_d2_null) 
predfit <- plot(simulationOutput_pred) 
predzi <- testZeroInflation(simulationOutput_pred) 
predod <- testDispersion(simulationOutput_pred)
plot(residuals(read_pred_u3))

all_pred <- u3_pred %>%
  left_join(d2_pred, by = c("sample" = "sample", "Method" = "Method", "Island" = "Island", "Year" = "Year", "ID" = "ID", "Date.Collected" = "Date.Collected", "dada2_sample" = "dada2_sample", "unoise_sample" = "unoise_sample", "Size" = "Size", "Extract..Date" = "Extract..Date", "Sterilized" = "Sterilized", "Source" = "Source", "surf_ster" = "surf_ster")) %>%
  gather(pipeline, reads, reads_u3, reads_d2) %>%
  group_by(Sterilized, pipeline) %>%
  summarise(mean = mean(reads), total = n(), se = sd(reads)/sqrt(total))

```

```{r pred graph, include=TRUE, echo=FALSE}
(predators <- ggplot(all_pred, aes(x = Sterilized, y = mean, fill = Sterilized)) +
  geom_bar(stat="identity", position= "dodge") +theme_bw() +
  labs(x = "Surface sterilization treatment", y = "Predator DNA reads",
       title = "Predator DNA read abundance by surface sterilization and algorithm") +
  scale_fill_manual(values = c("#7B8D65","#F29979")) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean +se), width = 0.2) +
    scale_y_sqrt() +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized")) + facet_wrap(~pipeline))
```

####2. Human DNA won't work as a line of evidence because it's only present in one sample:

```{r human DNA, include = TRUE}
#this binds them all to the community matrix for analyses
u3_comm_rare %>%
  left_join(u3_id_all, by = "ASV") %>% #combine with taxonomy DF
  filter(Order_ncbi == "Primates") %>% #select human
  dplyr::select(-ID_bold, -ID_ncbi) %>% #remove taxonomy columns
  group_by(ASV) %>% 
  gather(sample, reads, HEV07:HEV29, factor_key = TRUE) %>% #make long while grouped by ASV
  group_by(sample) %>%
  summarise(reads_u3 = sum(reads)) %>%#summarise total prey reads by sample 
  filter(reads_u3 > 0)
```

####3. What about the abundance of other prey items?

It does not significantly change with sterilization, but does seem to have a wider range with surface sterilization... *(could this be the one weird outlier?)*

```{r other prey DNA, include = TRUE, fig.show='hide'}
#this binds them all to the community matrix for analyses
u3_ot_prey <- u3_comm_rare %>%
  left_join(u3_id_all, by = "ASV") %>% #combine with taxonomy DF
  filter(taxonomy == "prey") %>% #select prey DNA only
  filter(!ID_ncbi == "Acrididae") %>% #don't choose known prey item
  dplyr::select(-ID_bold, -ID_ncbi) %>% #remove taxonomy columns
  group_by(ASV) %>% 
  gather(sample, reads, HEV07:HEV29, factor_key = TRUE) %>% #make long while grouped by ASV
  group_by(sample) %>%
  summarise(reads = sum(reads)) %>% #summarise total prey reads by sample 
  left_join(metadata, by = "sample") %>%
  mutate(pipeline = "u3")

#do same for Dada2

#this binds them all to the community matrix for analyses
d2_ot_prey <- d2_comm_rare %>%
  left_join(d2_id_all, by = "ASV") %>% #combine with taxonomy DF 
  filter(taxonomy == "prey") %>% #select prey DNA only
  filter(!ID_ncbi == "Acrididae") %>% #don't choose known prey item
  dplyr::select(-ID_bold, -ID_ncbi, -ID_nondiet) %>% #remove taxonomy columns
  group_by(ASV) %>% 
  gather(sample, reads, HEV07:HEV29, factor_key = TRUE) %>% #make long while grouped by ASV
  group_by(sample) %>%
  summarise(reads = sum(reads))  %>% #summarise total prey reads by sample 
  left_join(metadata, by = "sample") %>%
  mutate(pipeline = "d2")

#all_pred$surf_ster <- ifelse(all_pred$Sterilized == "NS", 0, 1)
library(glmmTMB)

#U3:
u3_ot_prey$surf_ster <- ifelse(u3_ot_prey$Sterilized == "NS", 0, 1)
read_prey_u3 <- glmmTMB(reads ~ surf_ster,
                    data = u3_ot_prey,
                 family = genpois)

read_prey_u3_null <- glmmTMB(reads ~ 1,
                    data = u3_ot_prey,
                    family = genpois)

AICc(read_prey_u3,read_prey_u3_null) 

simulationOutput <- simulateResiduals(fittedModel = read_prey_u3_null) 
fit <- plot(simulationOutput) 
zi <- testZeroInflation(simulationOutput) 
od <- testDispersion(simulationOutput)
plot(residuals(read_prey_u3_null))

#DADA2
d2_ot_prey$surf_ster <- ifelse(d2_ot_prey$Sterilized == "NS", 0, 1)
read_prey_d2 <- glmmTMB(reads ~ surf_ster,
                    data = d2_ot_prey,
                 family = genpois)

read_prey_d2_null <- glmmTMB(reads ~ 1,
                    data = d2_ot_prey,
                    family = genpois)

AICc(read_prey_d2,read_prey_d2_null) 

simulationOutput <- simulateResiduals(fittedModel = read_prey_d2_null) 
fit <- plot(simulationOutput) 
zi <- testZeroInflation(simulationOutput) 
od <- testDispersion(simulationOutput)
plot(residuals(read_prey_d2_null))

all_prey <- u3_ot_prey %>%
  bind_rows(d2_ot_prey) %>%
  group_by(Sterilized, pipeline) %>%
  filter(!sample == "HEV14") %>%
  summarise(mean = mean(reads), total = n(), se = sd(reads)/sqrt(total))
```

```{r other prey graph, include=TRUE, echo=FALSE}
(prey <- ggplot(all_prey, aes(x = Sterilized, y = mean, fill = Sterilized)) +
  geom_bar(stat="identity", position= "dodge") +theme_bw() +
  labs(x = "Surface sterilization treatment", y = "Other prey DNA reads",
       title = "Prey DNA read abundance by surface sterilization and algorithm") +
  scale_fill_manual(values = c("#7B8D65","#F29979")) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean +se), width = 0.2) +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized")) + facet_wrap(~pipeline))
```

####4. Raw sequencing depth assigned to ASVs (this should be equal if DNA *is not* degraded).

Raw sequencing depth (total number of sequences assigned to ASVs) *does* decrease with sterilization, lending credence to the DNA degradation hypothesis. We rarefied all our samples to the lowest sampling depth for this dataset prior to any analyses of detection, so I'm wondering that given all the other lines of evidence suggesting that surface sterilization does not seem to influence other pools of DNA in a negative way whether this is less important. Really, DNA that *is not* from our known prey is much less abundant in our samples and it does not go away with surface sterilization, meaning that both a) it is likely ingested DNA rather than surface contamination and 2) if this DNA does not disappear, then so should our known prey items not disappear.

```{r raw sequencing depth, include=TRUE, fig.show = 'hide'}
u3_depth <- u3_comm %>%
  dplyr::select(all_of(lab)) %>%
  gather(sample, reads, HEV07:HEV29) %>%
  group_by(sample) %>%
  summarise(reads = sum(reads)) %>%
  left_join(metadata, by = "sample") %>%
  mutate(pipeline = "u3")

d2_depth <- d2_comm %>%
  dplyr::select(all_of(lab)) %>%
  gather(sample, reads, HEV07:HEV29) %>%
  group_by(sample) %>%
  summarise(reads = sum(reads)) %>%
  left_join(metadata, by = "sample") %>%
  mutate(pipeline = "d2")

u3_depth$surf_ster <- ifelse(u3_depth$Sterilized == "NS", 0, 1)

depth_u3_mod <- glmmTMB(reads ~ surf_ster,
                    data = u3_depth,
                 family = truncated_genpois)

depth_u3_mod_null <- glmmTMB(reads ~ 1,
                    data = u3_depth,
                 family = truncated_genpois)

AICc(depth_u3_mod, depth_u3_mod_null) 

simulationOutput <- simulateResiduals(fittedModel = depth_u3_mod) 
fit <- plot(simulationOutput) 
predzi <- testZeroInflation(simulationOutput) 
predod <- testDispersion(simulationOutput)
plot(residuals(depth_u3_mod))

plot(allEffects(depth_u3_mod))
model.emm <- emmeans(depth_u3_mod, "surf_ster")
pairs(model.emm)

d2_depth$surf_ster <- ifelse(d2_depth$Sterilized == "NS", 0, 1)

depth_d2_mod <- glmmTMB(reads ~ surf_ster,
                    data = d2_depth,
                 family = truncated_genpois)

depth_d2_mod_null <- glmmTMB(reads ~ 1,
                    data = d2_depth,
                 family = truncated_genpois)

AICc(depth_d2_mod, depth_d2_mod_null) 

simulationOutput <- simulateResiduals(fittedModel = depth_d2_mod) 
fit <- plot(simulationOutput) 
predzi <- testZeroInflation(simulationOutput) 
predod <- testDispersion(simulationOutput)
plot(residuals(depth_u3_mod))

plot(allEffects(depth_d2_mod))
model.emm <- emmeans(depth_d2_mod, "surf_ster")
pairs(model.emm)

all_depth <- u3_depth %>%
  bind_rows(d2_depth) %>%
  group_by(Sterilized, pipeline) %>%
  summarise(mean= mean(reads), total = n(), se = sd(reads)/sqrt(total))
```

```{r raw sequenceing depth graph, include=TRUE, echo=FALSE}
(depth <- ggplot(all_depth, aes(x = Sterilized, y = mean, fill = Sterilized)) +
  geom_bar(stat="identity", position= "dodge") +theme_bw() +
  labs(x = "Surface sterilization treatment", y = "Total sequencing depth",
       title = "Sequencing depth by surface sterilization and algorithm") +
  scale_fill_manual(values = c("#7B8D65","#F29979")) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean +se), width = 0.2) +
    scale_y_sqrt() +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized")) + facet_wrap(~pipeline))
```


####Summary of alternative hypothesis explorations

**Predator and other prey DNA does not change with surface sterilization. **

**Raw sequencing depth decreases significantly for DADA2 but not UNOISE3 with surface sterilization.**


####5. Lack of significant changes in field-collected individuals may help build evidence for surface sterilization vs. DNA degradation patterns. Alternatively, may suggest that the ecological inference is unchanged whether the source of DNA disappearing is due to removing surface contaminants or because of the degradation of some DNA.

# Up next: **Part 2**: Environmental contamination for field-collected predators
