---
title: "DNA_Diet_Methods"
author: "Ana Miller-ter Kuile"
date: "3/11/2020"
output:
  word_document:
    toc: yes
  html_document:
    toc: yes
    toc_float:
      collapsed: no
---
# DNA Diet Methods Analyses

```{r packages required, include = FALSE}
library(ggplot2)
library(tidyverse)
library(here)
library(cowplot)
library(fitdistrplus)
```

## Motivation
This project creates a workflow for analyzing the diets of invertebrate predators using high throughput sequencing of gut contents. This method provides a promising way to get highly-resolved diet data from consumers across ecosystems. However, as these methods develop, there are challenges that vary by consumers and ecosystems.

These challenges can vary by environment, ecosystem, and consumer. Specifically for ecosystems where consumers are taxonomically similar to their prey and may come in contact with their prey items in the environment (here, invertebrate ecosystems, but one or both of these challenges can occur in other ecosystems as well), we face both a bioinformatics challenge and a sample processing challenge. 

1. **Bioinformatics challenge**: To extract all possible prey items from predator guts when predator and prey are taxonomoically similar, it is best practice to use a set of PCR primers that target all possible prey. However, a side effect of this is that these primers will also end up amplifying a large amount of predator DNA. As a result, these datasets are dominated by predator DNA, and so detecting relatively rare prey sequences in these datasets is key. As the molecular ecology field moves toward using amplicon sequence variants (ASVs) as biologically-real units of biodiversity in high throughput datasets, these types of datsets dominated by predator DNA are even more challenging since these ASV clustering pipelines use sequence abundance as a way to cluster sequences into similar, biologically-real groups of sequences. Therefore, any clustering pipeline used for DNA diet data dominated by the predator must 

  + detect prey sequences that are taxonomically similar to the DNA of predators and
  
  + detect prey sequences that are relatively rare compared to the DNA of predators

```{r ASV assignment figure, echo=FALSE, out.width="100%", fig.align='center', fig.cap = "Figure 1: Denoising pipelines (including DADA2 and UNOISE3) consider sequence abundance and error rate in raw sequence datasets when removing sequences that have high error and low abundance and combining their reads with those of their most similar neighbor sequence with high abundance and low error (amplicon sequence variants or ASVs). Because our dataset has high abundance predator DNA with lower abundance prey DNA, any denoising pipeline we use must accurately be able to distinguish low abundance prey ASVs from high abundance predator ASVs."}
knitr::include_graphics(here("Pictures", "denoise_picture.png"))
```
  
2. **Sample processing challenge**: Many invertebrate predators are very small, making extracting gut or fecal contents from the rest of the body of the consumer for DNA extraction prohibitive or impossible. In addition, these organisms are often in contact with many other parts of their environment, either through touching most of their body surface area to an environmental surface (i.e. ground, leaves, branches, etc), or through physically moving through these environments (i.e. burrowing). As a result, when extracting DNA from these organisms, we often must extract DNA from full bodies or from full body sections and can run the risk of 

  + contaminants on the outside surfaces of organisms drowning out DNA sequences from gut contents or 
  
  + falsely inflating prey richness through contact in the environment and not ingestion

```{r surface contamination, echo=FALSE, out.width="100%", fig.align='center', fig.cap = "Figure 2: DNA contaminating the outside surfaces of predators may alter ecological inference of diet diversity."}
knitr::include_graphics(here("Pictures", "surface_contam.png"))
```

We addressed both of these challenges in the following study. We assessed how different denoising/clustering pipelines perform with these types of datasets and provide a template for other studies interested in interactions between invertebrate predators and prey, but with the idea that the same sort of process could work well for other study systems (i.e. vertebrates that eat vertebrates) as well. We then used the best bioinformatics pipeline(s) across samples to compare the communities of reads/ASVs between surface-sterilized and non-surface sterilized individuals, assessing whether DNA one the outside of organisms is likely to mislead ecological inference. 

## The data

The data for this project consists of 56 individuals of the spider species *Heteropoda venatoria* from Palmyra Atoll National Wildlife Refuge. We collected 37 of these individuals in the field in 2015 and immediately froze them in a -80C freezer until processing in 2019. The remaining 19 individuals we collected in 2017, starved in mesocosms in the lab for 24 hours, and then fed a known diet item (the grasshopper *Oxya japonica*). After spiders had been in mesocosms with prey items and eaten them (12-24 hours), we froze them at -20C. Prior to final preservation in 80% EtOH, we surface sterilized 8 of these 19 individuals by submerging them in a 10% bleach to DI water solution for 2 minutes and then washing them in DI water for 2 minutes. We performed the same sterilization protocol on 18 of the 37 field-collected individuals after we had brought them to the lab from the field (in 2019). (see Methods document for full extraction, PCR, and sequencing protocols). 

```{r experimental setup, echo=FALSE, out.width="100%", fig.align='center', fig.cap = "Figure 3: Our surface-contamination experiment included four groups of varying sample sizes: both field-collected and lab-fed sterilized individuals and unsterilized individuals."}
knitr::include_graphics(here("Pictures", "experiment_setup.png"))
```

We ran all samples on an Illumina MiSeq platform at the UCSB Genetics Core. We cleaned low-quality sequences from this set of ~15 million sequences returned from the genetics core, and then used two clustering pipelines (DADA2 and UNOISE3) that use sequence abundance and quality to combine similar sequences (which are often different because of errors in amplification or sequencing and actually represent the same sequence) into clusters representing the most common real sequence. We performed each of these clustering pipelines twice: once on the complete list of sequences, and once on a set of sequences that had been "cleaned" of predator and non-diet sequences using the program BBSplit. After the denoising/clustering process, we are left with a matrix very similar to any community matrix, with columns corresponding to each sample (spider individual) and rows corresponding to ASVs (sequences) in that individual. In this way, each spider becomes functionally a "community" of species which can be treated very similarly to any other matrix of community data (i.e. in vegan with multidimensional analyses, etc). 

```{r an example matrix, echo=FALSE}
matrix_example <- read.csv(here("data", "ASV_tables", "dada2_uc_asv_tab.tsv"), sep= "\t")
matrix_example_small <- matrix_example[1:10,]
matrix_example_smaller <- matrix_example_small[c(1, 4:9)]
head(matrix_example_smaller, n=10)
```

## Outline
1. **Assess sequencing depth**, ensuring that our dataset is complete enough to answer any further questions

2. **Pipeline performance**, or: Does clustering method matter?

  + *Clustering specificity*: Positive and negative controls map to fewer ASVs
  + *Total ASVs (richness)*: A greater number of ASVs means a higher likelihood the full diversity of the sample has been detected
  + *Number of prey ASVs (diet richness)*: A higher diet richness means the pipeline better distinguishes prey ASVs from predator
  + *Phylogenetic diversity of prey ASVs*: A better pipeline will detect a broader phylogenetic diversity of prey
  + *Prey read abundance*: More prey reads means a higher likelihood of picking up rarer prey items
  + *Percent of ASVs that are prey*: This corrects for variable sequencing depth and a higher percentage means a higher likelihood that all prey are detected
  + *Percent of reads that are prey*: Again, correcting for sequencing depth, a higher percentage means a higher likelihood that all prey are detected
  + *Amount of known diet reads*: A better pipeline will be better at detecting a diet item we KNOW to be present in all samples
  
3. **Error checking** prior to looking at sterilization effects: is there error in our samples from sequence jumping?

4. **Sample processing**, or: Does sterilization matter?

  + *Presence*: does prey detection change with sterilization?
    - "false positives" of prey on the outside surfaces of predators?
    - contamination drowns out prey DNA?
    - contamination/false positives decrease detection of known prey items?
  + *Abundance*: does prey abundance change with sterilization?
    - Contamination decreases prey abundance?
    - Contamination decreases known prey abundance?
    
5. **Other questions** (i.e. things we haven't gotten to yet but am still thinking about)
  + Does cleaning step reliably remove predator and non-predator, non-prey sequences?
  + Do predator and prey read abundances track in a way that we can use?
  
## Want just the highlights? Jump to the [summary](#summary)

### 1. Sequencing Depth

We assessed sequencing depth to ensure that we had captured the full diversity of samples. We used an approach by Chao et al. (2016) using interpolation and extrapolation in the iNEXT package (version 2.0.20) in R (version 3.6.1). We computed a sampling effort curve of the number of species (ASVs) discovered in each sample compared to the number of sequence reads in that sample. These sampling effort curves illustrate sampling coverage (SC), which is the percentage of species (ASVs) predicted to be in that sample based on the sampling effort.

```{r sequencing depth setup, include=FALSE}
source(here("8_statistical_analyses", "sampling_depth.R"))
```

```{r sequencing depth graph, echo=FALSE, warning=FALSE}
seq_depth_graph
```

**Sequencing Depth Summary:**

Visually, it looks like we did a good job of sequencing the full diversity of samples. We can also look at the sampling coverage values for all samples:

```{r sequencing depth}
seq_depth$DataInfo$SC
```

And it looks like we have captured 99%+ of the diversity of every sample!

### 2. Pipeline performance, or: Does clustering method matter?

```{r pipeline perfomance setup, include=FALSE}
source(here("8_statistical_analyses", "pipeline_asv_sum_stats.R"))
```

We looked at a bunch of different ways the pipelines performed both bioinformatically and ecologically. If you don't want to spend time on each one, you can jump to the pipeline performance [summary table](#pipelines), which provides the scores of each pipeline in each measure of performance. 

#### a. Clustering specificity (positive and negative control ASVs)

Different clustering methods may have more or less specificity in their clustering. We checked for this by running several positive control samples and one negative control sample and looking at how different clustering pipelines assigned ASVs to these controls. The positive controls are cloned fungal species which Austen uses in his work. Because they are clones, they should map to one or very few ASVs. The negative controls were samples we ran through PCRs with no DNA in them and quantified to zero before submitting. Therefore, these should map to zero ASVs. **Takeaway**: smaller is better for both positive and negative control ASV counts. 

```{r control ASV counts, echo=FALSE}
control_count_plot
```

It seems that, first of all, cleaning removes all ASVs for the controls, which we would expect, since we filtered these based on known prey and predator ASVs. Furthermore, it seems that DADA2 does a better job of assigning fewer ASVs to positive controls than UNOISE3. Furthermore, the negative control had zero reads assigned to any reads in DADA2, while UNOISE3 assigned one ASV a value of 1 read for the negative control. 

**Winner: Either cleaned or DADA2 uncleaned**

#### b. Total ASVs (richness)

A good clustering pipeline will be able to pick up all the diversity in a dataset. In this case, a higher number of total ASVs (roughly, species richness) means better clustering method, since we want to capture the greatest diversity possible. However, we will go into some more specific measures of richness (prey richness) next, since just the total number of ASVs may not actually be prey richness, but may instead be the pipeline clustering more predator or non-prey ASVs. 

We looked at both total ASVs produced by each pipeline as well as the number of ASVs each clustering pipeline assigned to each sample. 

```{r total ASVs, echo=FALSE}
total_asv_graph
```

The uncleaned datasets produce more ASVs, and DADA2 produces more total ASVs than UNOISE3.

**Winner: DADA2**

When we look at the total number of ASVs produced per sample, we see a different pattern, however: 

```{r total ASVs per sample, echo=FALSE}
ASVs_sample_plot
```

Indeed, these differences among pipelines are statistically significant based on a random effects model with a full model structure of (comparing it to a null without pipeline, and then estimating marginal means between each pair):

```{r total ASV model, eval=FALSE}
tot_ASV_mod <- glmmTMB(value ~ pipeline + (1|sample),
                       data = by_sample_long,
                       family = "genpois")
```

With significant pairwise differences between all pipelines:

```{r marginal means of total ASV model, echo=FALSE}
pairs(model.emm_totASV)
```

The takeaway here is that both UNOISE3 unclean and clean produce more ASVs than their DADA2 counterparts.

**Winner: UNOISE3 uncleaned and cleaned**

#### c. Number of prey ASVs (diet richness) 

A higher total number of ASVs does not necessarily mean that a clustering pipeline is the best option, since these newly discovered ASVs may not actually be ASVs of interest, but rather ASVs of predators or other non-prey items (contamination, for example). Therefore, a better measure of whether a pipeline works best for the purposes of this study is to look at how well it clusters prey ASVs. We did this by combining taxonomic data to our ASV list, and assigning broad taxonomic categories to these ASVs. We combined taxonomic data from both GenBank (using BLAST and MEGAN) and from the BOLD IDEngine database. We combined taxonomic assignments from both sources and removed any assignment which did not match between these two databases. Anything that mapped to the predator Family, Genus, or Species was given a taxonomic category of "predator", anything that could possibly be prey (including arthropods and vertebrates in this dataset) were given a taxonomic category of "prey". Anything that was not given a specific taxonomic assignment but which could have been prey or predator was given the category of "unknown" (i.e. a BLAST taxonomic assignment of "Arthropoda"). There were other items in the dataset that were definitely not predators or prey (mostly fungi) and these were given a taxonomic assignment of "non-prey". Our final category included any ASV that was clustered but which was not assigned a taxonomy; this category was "no hit". 

We then subset just the prey ASVs from this dataset and compared the per sample number of prey ASVs assigned by each pipeline, which is visualized here:

```{r prey ASV graph, echo=FALSE}
prey_ASV_graph
```

We again looked for significant across group differences in prey ASVs using a mixed model (comparing it to a null without pipeline, and then estimating marginal means between each pair):

```{r prey ASV model, eval=FALSE}         
lme_prey_ASVs <- glmmTMB(ASVs ~ pipeline + (1|sample),
                 data = taxa_ASVs_prey,
                 family = "genpois")
```

And looked at pairwise differences between these groups: 
 
```{r marginal means of prey ASV model, echo=FALSE}                
pairs(model.emm_preyASV)
```

What we see is that cleaning does not significantly increase the number of ASVs assigned to prey items for either DADA2 or UNOISE3. However, what we see is that UNOISE3 assigns more prey ASVs to each sample, suggesting that UNOISE3 is better at detecting a greater prey richness than DADA2

**Winner: UNOISE unclean**

#### d. Phylogenetic diversity of prey ASVs

Not only might one pipeline be better at detecting total prey richness, but might also be able to cluster a broader phylogenetic diversity of prey items. This might be especially important in systems where predators are generalists and feed on organisms across broad taxonomic groups. Because the number of ASVs assigned to prey per sample is fairly low (less than 10 per sample) and because we are interested in just detection here and not necessarily abundance variations among samples, we will just qualitatively assess Faith's PD for each pipeline, summing the prey ASVs for each pipeline. Furthermore, we will combine phylogenetic analyses at the Order and Family level, since we have good taxonomic representation at these levels (15 orders, 19 families)

```{r order level phylogenetic diversity, echo=FALSE}
plot_grid(order_pd_graph, order_phy_graph, align="h")
```

UNOISE3 unclean picks up the greatest phylogenetic diversity. It is important to note that this is in spite of the fact that order richness is greater in the DADA2 dataset. UNOISE3 unclean has the lowest phylogenetic diversity, which is probably because we trained the cleaning program with DADA2 and some of the phylogenetic diversity of the UNOISE3 unclean dataset was removed. Based on the phylogenetic tree, DADA2 picked up Lepidoptera (moths and butterflies) and Sarcoptiformes (mites), while UNOISE3 picked up Squamata (geckos). 

```{r family level phylogenetic diversity, echo=FALSE}
plot_grid(family_pd_graph, family_phy_graph, align ="h")
```

Again, UNOISE3 unclean picks up the greatest phylogenetic diversity. It is important to note that this is in spite of the fact that, again, family richness is greater in the DADA2 dataset. UNOISE3 unclean has the lowest phylogenetic diversity, which is probably because we trained the cleaning program with DADA2 and some of the phylogenetic diversity of the UNOISE3 unclean dataset was removed. DADA2 picks up Formicidae (ants) and Suidasiidae (mites), while UNOISE3 picks up Xiphydridae (wasps) and Gekkoniidae (geckos). 

**Winner: UNOISE unclean**

#### e. Prey read abundance

Similar to the number of ASVs assigned to prey, the number of sequence reads assigned to each of these ASVs is important because rarer prey items are more likely to be detected the greater abundance each prey ASV is in each sample. 

We can visualize the number of reads of each ASV in each sample:

```{r prey reads graph, echo=FALSE}
prey_reads_graph
```

And again, we ran a mixed model to determine whether there are significant among pipeline detections in prey reads (comparing it to a null without pipeline, and then estimating marginal means between each pair):

```{r prey reads model, eval=FALSE}
read_mod <- glmmTMB(reads ~ pipeline + (1|sample),
                 data = taxa_reads_prey,
                 family = "genpois")
```

```{r marginal means for prey reads model}
pairs(model.emm_preyreads)
```

Cleaning did not significantly increase prey read abundance. UNOISE3 produced more prey reads per ASV per sample than DADA2

**Winner: UNOISE3 unclean**

#### f. Percent of ASVs that are prey

So far, we have been assuming that the total number of ASVs is independent of sequencing depth, which we know is not the case. Therefore, we ran the above ASV model again correcting for sequencing depth (using an offset of the total number of ASVs). These data can then be interpreted as the percent of all ASVs assigned to prey. 

```{r prey prop ASV graph, echo=FALSE}
prey_prop_graph
```


```{r prey ASV prop model, eval=FALSE}
prey_prop_mod <- glmmTMB(prey ~ pipeline + (1|sample),
                          data = ASV_totals_samples,
                          family = "genpois",
                         offset = log(tot))
```

```{r marginal means of prey ASV prop model, echo=FALSE}
pairs(model.emm_preyprop)
```

Unsurprisingly, cleaning increases the total proportion of prey ASVs in each sample. DADA2 has a higher proportion of prey ASVs than UNOISE3 for both clean and uncleaned datasets.

**Winner: DADA2 clean**

#### g. Percent of reads that are prey

Again, we have been assuming that the total number of reads is independent of sequencing depth, which we know is not the case. Therefore, we ran the above reads model again correcting for sequencing depth (using an offset). These data can then be interpreted as the percent of all reads assigned to prey. 

```{r prey read proportion graph, echo=FALSE}
prey_rprop_graph
```


```{r prey read proportion model, eval=FALSE}
prey_rprop_mod <- glmmTMB(prey ~ pipeline + (1|sample),
                         data = read_totals_samples,
                         family = "genpois",
                         offset = log(tot))
```

```{marginal means of the prey read proportion model, echo=FALSE}
pairs(model.emm_preyrprop)
```

Cleaning increased the total proportion of reads that are assigned to prey ASVs. There is no difference in the proportion of prey reads in samples for either DADA2 or UNOISE3. 

**Winner: cleaned datasets**

#### h. Amount of known diet reads

We know we fed lab organisms a specific diet item (*Oxya japonica*), and so we should trust the pipeline that detects this prey item better.

Let's look at detection across samples of this prey item (we subset our dataset to just be those that we fed the item to):

```{r known prey plot, echo=FALSE}
known_prey_plot
```

Again, we built a mixed model that we compared to a null model: 

```{r known prey model, eval=FALSE}
known_reads_mod <- glmmTMB(value ~ pipeline + (1|sample),
                          data = known_reads_long,
                          family = "genpois")
```

And looked at pairwise differences:

```{r marginal means known prey model, echo=FALSE}
pairs(model.emm_knownprey)
```

Cleaned datasets have significantly more reads of the known prey items per sample for DADA2, however cleaning did not increase detection of known prey for UNOISE3. UNOISE3 has better detection of known prey items. 

## Pipeline Performance Summary {#pipelines}

Compiling the results from this, we get a summary of all the measures of pipeline performance. In the case of a non-significant difference between pipelines, the pipeline with the most parsimonious protocol (i.e. no cleaning) will win. In the case that the parsimony is the same between pipelines, none is considered to be outperforming. (in the following table, an "X" designates the best performing pipeline, either by significant difference between groups or because it is the most parsimonious of a tie, "NS" signifies that these pipelines performed equally well to the winning, parsimonious pipeline, and "TIE" signifies that these pipelines performed equally well and have the same parsimony.)

```{r summary table of pipelines, echo=FALSE}
row1 <- c("Measure/Pipeline", "DADA2 UC", "UNOISE3 UC", "DADA2 C", "UNOISE C")
row2 <- c("Positive control ASVs", "X", "", "NS", "NS")
row3 <- c("Negative conotrol ASVs", "X", "", "NS", "NS")
row4 <- c("Total ASVs", "X", "", "", "")
row5 <- c("ASVs per sample", "", "X", "", "NS")
row6 <- c("Prey ASVs", "", "X", "", "")
row7 <- c("PD - Order", "", "X", "", "")
row8 <- c("PD - Family", "", "X", "", "")
row9 <- c("Prey reads", "", "X", "", "")
row10 <- c("Prey ASV %", "", "", "X", "")
row11 <- c("Prey read %", "", "", "TIE", "TIE")
row12 <- c("Known Diet", "", "X", "", "NS")

pipeline_table <- as.data.frame(rbind(row1, row2, row3, row4, row5, row6, row7, row8, row9, row10, row11, row12))
colnames(pipeline_table) <- pipeline_table$row1
knitr::kable(pipeline_table[2:12, ], col.names = row1, row.names=FALSE)

```

From this, we see that in most cases, the uncleaned pipelines perform equally well or better than the cleaned pipelines. So cleaning does not add much to our bioinformatic performance nor our ecological inference from this dataset. As a result, for the following analyses of sterilized vs. unsterilized individuals (both field and lab), we will be looking at the performance of DADA2 and UNOISE3 unclean. we will be looking at both pipelines because, even though UNOISE3 outperformed DADA2 ecologically, DADA2 outperformed UNOISE3 bioinformatically, so it is hard to choose a "winner", since neither performed best both ecologically and bioinformatically.

### 3. Error analyses from sequence jumping

Before running community analyses, we first assessed sequencing error based on the number of sequences assigned to the negative control in each pipeline. Non-zero values in a negative control either represent contamination from the PCR process or from the sequencing platform. Because we ensured the concentration of our negative control was zero (or near-zero) prior to sequencing submission, we assume that any non-zero values in the negative control likely come from the sequencing platform and represent the rate at which sequences jump from sample to sample across all samples in the run. Therefore, we used this value to build an error distribution and correct our raw sequencing data prior to community analyses. To do this, we fit the ASV abundances for the negative control to a poisson and negative binomial distribution and used BIC to fit the best of the two distributions to the data (CITE JERDE HERE). We then used the distribution of the best-fitting model to predict whether values of different abundances were likely due to error or not (if significant p-value, i.e. 0.001 or lower, these values correspond to real biological diversity and not to sequencing error). Any values which are likely due to sequencing error were removed prior to community analyses. 

```{r error analysis setup, include=FALSE}
source(here("7_error_checking", "asv_table_cleaning.R"))
```

During denoising via DADA2, the negative dropped out as having zero reads in any ASVs, so for this dataset, everything that is present is likely not to come from error. For UNOISE3, there is only one ASV with a non-zero read value in the negative (and it has 1 read assigned), so we will do an error check for this dataset. Basically, we will be asking whether any read of 1 is sequencing error or whether it represents real biological diversity.

```{r hist of error in negative, echo=FALSE}
hist(error_check$NEG)
```

This dataset may fit a poisson, negative binomial, or a zero-inflated version of either of these models. We started by fitting a poisson and negative binomial model and comparing via BIC

```{r poisson and NB of error check, eval=FALSE}
model1 <- glm(NEG ~ 1, family = poisson, data = error_check)
model2 <- glm.nb(NEG ~ 1, data = error_check)
```

```{r BIC of poisson and NB of error, echo=FALSE}
BIC(model1, model2)
```

Model one is the better of the two, but is it a good fit? We used the DHARMa package to evaluate model fit, in particular whether this dataset was zero-inflated.

```{r model fits, echo=FALSE}
plot(simulate_poi, asFactor=TRUE)
testZeroInflation(simulate_poi)
```

The QQ plot looks okay and the residuals look... as good as they probably can given that we basically have a dataset of zero. Additionally, this dataset is not considered zero-inflated, so the poisson can be assumed to be the best model of this dataset. 

With this model, we can now predict whether reads of different values fit this distribution (i.e. represent error) or whether they do not fit into this distribution. Therefore, when interpreting p-values of values predicted given this distribution, a significant p-value indicates that the value is likely to come from a significantly different distribution than the distribution of the error, and a non-significant p-value means that the value is not significantly different from the values in that distribution (and so represents error).

```{r fit and predict distribution}
fitdistr(error_check$NEG, "Poisson")
```

From this, we get a lambda value of 0.005681818, which we can then use to predict fits of different values given this distribution and lambda. We will start with our lowest read value (1):

```{r fit values to predict error distribution}
ppois(1,lambda=0.005681818,lower=FALSE)
```

Since we get a significant p-value from this prediction (p < 0.001), a value of 1 in our dataset can be assumed to *NOT* come from our error distribution. Therefore, any value of 1 or greater represents real biological diversity in our samples, and we do not have to correct for error in our datasets. Onward!

### 4. Sample processing, or: Does sterilization matter?

#### a. Presence, or does prey detection change with sterilization?

Contamination on the outside of individuals can have various consequences, including:
1. False positives: potential prey DNA on the outside surface of predators falsely inflates prey diversity estimates.
2. Prey dilution: contamination on the outside surface of predators dominates and drowns out prey DNA sequences.
  + Within this dilution category, we can consider the lab-fed spiders as a check of this idea: if contamination or false positives are in play, then they may decrease the detection of the DNA coming from known prey items. 

```{r presence analyses source, include=FALSE}
source(here("8_statistical_analyses", "presence-based_comm_diss.R"))
```

We first looked at presence/absence communities across sterilized and unsterilized individuals. Because we were interested in different ecological questions for field and lab individuals (and because we stored these samples differently prior to extraction, which can alter DNA extraction outcomes), we compared these communities separately. Additionally, we chose to assess both communities created by UNOISE3 and DADA2, and becuase these created different sets of ASVs and taxonomies, we also assessed these communities separately from each other. We ended up with roughly between 175-225 ASVs for these pipelines, however, some of these, once we compared them to GenBank and BOLD, matched to the same species (i.e. several mapped to the predator). Therefore, prior to any community analyses, we combined all the reads from each shared taxonomy. We did this for only items that could be prey (i.e. not fungi and other non-animals). For anything that was definitely not prey, we gave a category of "non-prey". For anything with too high a taxonomic assignment to be given a category of predator, prey, or non-prey (i.e. something that mapped to an "Arthropoda" or "Opisthokonta" taxonomy only), we gave a category of "unknown", since they could be predator, prey, or neither. 

For the field samples run through UNOISE, some example unique taxonomies are:

```{r example taxonomy, echo=FALSE}
head(unique_field_u3)
```

Combining taxonomies greatly simplified each community for ease of analysis. 

```{r summary table of uniques, echo=FALSE}
row1_2 <- c("", "UNOISE3-Field", "DADA2-Field", "UNOISE3-Lab", "DADA2-Lab")
row2_2 <- c("Unique species", "28", "31", "17", "14")

unique_table <- as.data.frame(rbind(row1_2, row2_2))
colnames(unique_table) <- unique_table$row1_2
knitr::kable(unique_table[2, ], col.names = row1_2, row.names=FALSE)
```

We ran a PERMANOVA for each of these piplines (UNOISE3-Field, DADA2-Field, UNOISE3-Lab, DADA2-Lab) both with a glmm and with the adonis() function in vegan. 

```{r example PERMANOVA model, eval=FALSE}
#a full glmm which we compared to a null model without sterilization
u3_field_model <- glmer(Presence ~ Ster + (1+Ster|species_ID), 
                     data = u3_field_glmm,
                     family = "binomial")

#and with ADONIS
adonis(fld_u3_nmds ~ Sterilized, data = u3_meta_field, dist = "jaccard", binary=TRUE)
```

The takeaways of the results (which we will illustrate in plots/outputs below)

1.	Neither field or lab communities are significantly different between sterilized and unsterilized individuals for either UNOISE3 or DADA2. 

2. For field samples, when we look at the per-species differences between sterilized and unsterilized individuals, in general for both UNOISE3 and DADA2, some species are slightly more abundant on indiviuals that are *not* surface sterilized than those that are. These include "predator"" DNA, "non-diet", "unknown", and "no taxonomy". Additionally, two possible diet items only categorized to high taxonomy appear in the DADA2 dataset as more abundant on non-sterilized individuals ("Neoptera", and "Araneae"). Several possible diet items appear more abundantly on non-sterilized individuals, including *Scholastes lonchifer* (fly), *Isometrus maculatus* (scorpion), unknown Ectobiidae roach, and *Anisolabis maritima* (earwig). Interestingly, only the Ectobiidae roach and *A. maritima* (earwig) are more likely to be present in non-sterilized individuals across both DADA2 and UNOISE3. While these species were predicted to be more present when non-sterilized, their predicted presence is only 2% (± 7% SE) and 6% (SE ± 7%) greater on non-sterilized than sterilized for each UNOISE3 and DADA2, respectively. 

3. For lab samples, UNOISE3 resulted in no species-specific differences between sterilized and unsterilized individuals. For DADA2, some species were more likely to be found on unsterilized individuals, including the predator, the known diet item (*Oxya japonica*), and non-diet items. The predicted presence was 3% (± SE 4%) higher on unsterilized individuals. 
  + An interesting note here is that the known prey item, *O. japonica*, is actually *more* likely to be observed in unsterilized individuals.

Field sample communities:

```{r field community NMDS for both pipelines, echo=FALSE}
plot_grid(u3_f_nmds_graph, d2_f_nmds_graph, align = "h")
```

And adonis() PERMANOVA: 

```{r adonis models for field samples}
adonis(fld_u3_nmds ~ Sterilized, data = u3_meta_field, dist = "jaccard", binary=TRUE)

adonis(fld_d2_nmds ~ Sterilized, data = d2_meta_field, dist = "jaccard", binary=TRUE)
```

By-species differences based on GLMER PERMANOVA:

```{r dotplots of random effects for field models, echo=FALSE}
plot_grid(u3f_re_plot, d2f_re_plot, align = "h")
```

Lab sample communities: 

```{r lab community NMDS for both pipelines, echo=FALSE}
plot_grid(u3_l_nmds_graph, d2_l_nmds_graph, align = "h")
```

And adonis() PERMANOVA:

```{r adonis models for lab samples}
adonis(lab_u3_nmds ~ Sterilized, data = u3_meta_lab, dist = "jaccard", binary = TRUE)

adonis(lab_d2_nmds ~ Sterilized, data = d2_meta_lab, dist = "jaccard", binary = TRUE)
```

By-species differences based on GLMER PERMANOVA:

```{r dotplots of random effects for lab models, echo=FALSE}
plot_grid(u3l_re_plot, d2l_re_plot, align = "h")
```

  + The quick note above that known prey (*Oxya japonica*) is more likely to be picked up on unsterilized individuals than sterilized individuals is illustrated below. For all percentages, the total number of samples/individuals is 19, and the maximum with absenct known diet item is 3 samples.
  
```{r known diet presence by sterilization, echo=FALSE}
absent_plot
```

**Returning to our predictions**
1. False positives: potential prey DNA on the outside surface of predators falsely inflates prey diversity estimates.

  + This does not seem to be a concern for our dataset.
  
2. Prey dilution: contamination on the outside surface of predators dominates and drowns out prey DNA sequences.

  + Again, this does not seem to be a concern for our dataset.
  
3. Within this dilution category, we can consider the lab-fed spiders as a check of this idea: if contamination or false positives are in play, then they may decrease the detection of the DNA coming from known prey items. 

  + Detection DECREASES with sterilization (maybe because of DNA on outside of predator because of being in the mesocosms?)
  
#### b. Abundance, or: does prey abundance change with sterilization?

After we looked at prey detection through presence analyses, we also looked a little deeper at whether prey and known prey abundance changes with sterilization. The predictions here are similar to those of detection, but also suggest a mechanism - if some things are not detected, it may be that their abundances are low and they get drowned out by contamination. Our predictions here are that:

1. Surface contamination could decrease prey abundance by saturating PCR or the sequencing run before rare diet items are amplified or sequenced. 

2. As a sub-category here, we can also predict that surface contamination could decrease known prey abundance for the same reasons. 
```{r abundance analyses source, include=FALSE}
source(here("8_statistical_analyses", "abund-based_comm_diss.R"))
```

We approached abundance-based community analyses similarly to presence-based analyses. We added an additional step of rarefying our samples before analyses for abundance analyses since sequencing depth can vary across samples. We wanted to ensure that we corrected for this variation in sampling depth before drawing ecological meaning from abundances in these samples. Furthermore, after rarefying, we combined all reads by broader taxonomic level, because we were interested in whether contamination (either of non-prey or prey DNA) would influence the abundance of both possible prey items and also known prey items (for lab-fed individuals). Therefore, we combined all species in the field dataset into five categories: "predator", "prey", "non-diet", "no hit", and "unknown" with the same distinctions as for the presence-based analyses. For lab-fed individuals, we added an additional category, so we had: "predator" "non-diet", "known prey" (*Oxya japonica*), "unknown", "other prey", and "no hit".

After combining reads, we again performed PERMANOVA analyses by sample type (field or lab) and pipeline (DADA2 or UNOISE3). 

The takeaways of the results (which we will illustrate in plots/outputs below)

1. Communities are similar across sterilized-unsterilized for both DADA2 and UNOISE3 lab and field individuals. 

2. The only group for which there were significantly non-zero differences between sterilized and unsterilized was for UNOISE3 Lab. For this analysis, the abundance of known prey decreased with sterilization. The same trend, though not significantly non-zero is true for the DADA2 lab data. The abundance of known prey reads decreased from an average of 100 per sample to 36 per sample for UNOISE3 and from 91 to 36 for DADA2.

Abundance-based communities for field samples: 

```{r abundance dissimilarity field, echo=FALSE}
plot_grid(bray_u3_fld_grph, bray_d2_fld_grph, align = "h")
```

Adonis PERMANOVA outcomes:

```{r adonis field abundance}
adonis(u3_fld_adonis ~ Sterilized, data = u3_meta_field, dist = "bray")

adonis(d2_fld_adonis ~ Sterilized, data = d2_meta_field, dist = "bray")
```

Random effects, showing no significant by-group differences:

```{r random effects field abundance, echo=FALSE}
plot_grid(u3f_re_plot, d2f_re_plot, align = "h")
```

Abundance-based communities for lab samples: 

```{r abundance dissimilarity lab, echo = FALSE}

plot_grid(bray_u3_lab_grph, bray_d2_lab_grph, align ="h")

```

Adonis PERMANOVA outcomes:

```{r adonis lab abundance}
adonis(u3_lab_adonis ~ Sterilized, data = u3_meta_lab, dist = "bray")

adonis(d2_lab_adonis ~ Sterilized, data = d2_meta_lab, dist = "bray")
```

Random effects, showing an decrease in known prey with sterilization for UNOISE3, and a similar trend for DADA2:

```{r random effects lab abundance, echo=FALSE}
plot_grid(u3l_re_plot, d2l_re_plot, align = "h")
```

**Returning to our predictions:**
1. Surface contamination drowns out prey abundance?

  + This does not seem to be a concern for our dataset.
  
2. Surface contamination drowns out known prey abundance?

  + The abundance of known prey actually decreases with sterilization, probably as a result of removing DNA from the outside of predators as they came into contact with prey items in mesocosms

### 5. Other questions, (i.e. things we haven't gotten to yet but am still thinking about)

1. Does cleaning step reliably remove predator and non-predator, non-prey sequences? 

  + Anecdotally, yes, but haven't analyzed yet.

2. Do predator and prey read abundances track in a way that we can use?

## Summary {#summary}

1. **Bioinformatics challenge**: DADA2 did a better job of assigning cloned positive controls to fewer ASVs, but UNOISE3 outperformed DADA2 in every ecologically significant analysis (summarized in [this](#pipelines) table). Cleaning datasets with BBSplit did not significantly increase the performance of either pipeline. While we used both for further analyses, because UNOISE3 did a better job of ecological measures and also did a good job with positive controls (assigning each to just 3 ASVs), **UNOISE3 does seem to be a better pipeline.**

Both pipelines successfully:

  + detect prey sequences that are taxonomically similar to the DNA of predators and
  
  + detect prey sequences that are relatively rare compared to the DNA of predators
  
2. **Sample processing challenge**: Surface contamination did not significantly alter overall community composition of either field-collected or lab-fed predators. Surface sterilization does seem to remove non-specific and non-diet items by a small fraction (less than 10%) in field-collected samples (and potentially lab-fed samples as well). Several potential prey items are very slightly more abundant (again, less than 10%) on unsterilized individuals from the field, which may suggest these "prey" items are potential contamination. If these are, indeed, contamination, they represent a 7 or 15% increase (DADA2 and UNOISE3, respectively) in the estimated diet richness of this predator (2/30 for DADA2 and 4/27 for UNOISE3). Surface sterilization actually *decreased* the presence and abundance of known prey in lab-fed predators. This finding may be important for future studies of diet DNA in controlled environments like mesocosms - care needs to be taken to ensure that detected prey DNA is *inside* the predator versus on the *outside* from predator and prey coming into contact with each other in their environment.

Surface sterilization shows that

  + contaminants on the outside surfaces of organisms do not drown out DNA sequences from gut contents (but care needs to be taken when prey and predator come in contact, as in the mesocosms, as this can falsely inflate the amount of DNA assigned to prey that is contamination from contact) and
  
  + Surface contamination seems to potentially mildly inflate prey richness (7 to 15%) through contact in the environment and not ingestion. Whether this inflation is ecologically significant is worth consideration. (I'm emailing with Chris Jerde at MSI about this topic currently.)