---
title: "Chapter 2: Diet Methods"
author: "Ana Miller-ter Kuile"
date: "4/22/2020"
output:
  html_document:
    theme: readable
    code_folding: hide
    toc: false
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r load packages}
library(here) #tidy data
library(tidyverse) #tidy data
library(ggplot2) #visualize data
library(emmeans) #marginal means of models
library(rcompanion) #pairwise comparisions
library(effects) #dotplots and stuff
library(lattice) #again dotplots
library(DHARMa) #model diagnostics
library(MuMIn) #model diagnostics
library(glmmTMB) #mixed models
library(lme4) #mixed models
library(ggeffects) #ggpredict function
library(performance) #for binomail model fit
library(see) #for binomial model fit
library(coin) #wilcoxon test
library(MASS) #glm.nb
library(vegan) #rrarefy
library(cowplot) #plot grid at end
#library(picante) #phylogenetic analyses
#library(ape) #phylogenetic analyses
#library(phytools) #phylogenetic analyses
#library(phyloseq) #phylogenetic analyses
library(hciR) #for as_matrix function
library(esc) #effect sizes
library(effsize) #alternative effect size package
library(egg) #inset graph
```

# Chapter 2: Do invertebrate consumers need to be surface sterilized before DNA metabarcoding diet analyses? 

# {.tabset .tabset-pills .tabset-fade}

### Explore the tabs for an introduction to the project, in-depth analyses for predator diets from contained and field environments, and view the summary of ecological interpretation. {#top}

## Introduction

**Target Journal:**
Methods journal (i.e. Molecular Ecology Resources, Methods in Ecology and Evolution, PLoS Methods paper)

**Baby:** 
We are trying to understand feeding interactions in complex systems (i.e. food web interactions) and their relationship to important ecological functions and processes (stability, etc.) (broader "why we would want to determine feeding interactions in ecosystems framing here"). 

**Werewolf:** 
We can use DNA metabarcoding to detect interactions that we’ve never had before that could address these challenges/questions. Including - diets of consumers too small to dissect for feeding interactions (cite Wirta, fungus beetle, etc.), fecal pellets of consumers to get long temporal scales and holistic diets (Kartzinel, others). While the metabarcoding approach is becoming more common, there are challenges with creating common practices across multiple systems, and knowing how to treat samples prior to DNA extraction and sequencing. 
Specifically: For many communities where metabarcoding could provide the most new information (ie. invertebrate-interactions where traditional diet analyses vs. gut contents is challenging/unfeasible) small body size has historically been the limitation to broad-scale diet analyses. DNA metabarcoding allows us to extract diet data from many of these consumers for the first time. However, even with DNA metabarcoding, it is necessary to extract DNA from entire individuals rather than targeting the gut contents of these organisms (again, because of small size). This means that while DNA metabarcoding can provide us with diet data, DNA contaminants on the outsides of individuals may alter prey diversity by falsely inflating estimated prey diversity (if prey DNA is on the outside surfaces of predators) or by drowning out prey DNA diversity because contaminant DNA will fill up sequencing runs. The possibility of contamination may vary by environment, where environments where consumers share substrate or confined spaces with prey (i.e. experimental environments, mesocosms, small environments such as bromeliads, small pools, soil, etc. (COME UP WITH EXHAUSTIVE LIST)) and so the choice to surface sterilize consumers prior to DNA extraction may be a choice made depending on the context. 

```{r surface contamination, echo=FALSE, include=TRUE, out.width="100%", fig.align='center', fig.cap = "Figure 1: DNA contaminating the outside surfaces of consumers may alter ecological inference of diet. This may be especially important in contained environments where predators and prey may come into contact with each other or shared surfaces."}
knitr::include_graphics(here("Pictures", "surface_contam.png"))
```

**Silver Bullet:**
We look at prey detection, richness, and DNA read abundance in invertebrate consumers, and compare these metrics between predators that we have surface sterilized (thus attempting to remove surface contamination) to those we have not surface sterilized. We have predators from two environments: 1) an experimental environment where predators were fed known prey items in close contact with those same prey items, and 2) a field environment where predators fed on natural prey items and their contact with prey in the environment was unknown. 

  + Experimental/Contained: Or, organisms with a known range of diet items in which we are trying to detect known diet. 

    - Does surface sterilization of predators from these environments alter the detection and abundance of known prey? 

  + Field: Or, for organisms with unknown diet in which we are trying to maximize prey DNA recovery and verifying that “prey” DNA is, indeed, ingested prey and not surface contamination. 

    - Big picture: Does surface sterilization alter prey detection, diversity, and read abundance?

    - Zooming in: Does surface sterilization change the species composition of prey DNA, thus suggesting that some “prey” DNA may be surface contamination?

## Contained Environment

### Contained Environment Analyses (Mesocosms)

This code examines the diets of mesocosm predators (the spider *Heteropoda venatoria*), specifically looking for the detection and abundance of prey DNA in predators which we have fed a known diet item (*Oxya japonica*). 

Specifically, we are exploring:

#### Does surface sterilization alter prey ASV detection or read abundance by removing surface contamination?

### Why it matters

Contained environments (both natural and artificial) are regularly used as model systems for understanding the effects of consumptive interactions on community structure (CITE mesocosms AND pools, or similar contained environments here). DNA metabarcoding offers a new way to confirm the effects of predator consumption in these environments, and, for example, separate consumption- vs. fear-based functional responses to community structure (Shmitz CITE), and maybe even get at numerical/biomass consumption (CITE the stuff from Mahon about intra-specific gene variation). In any of these environments, predators and prey may come into contact with each other in their environment through shared surfaces or substrates. When using DNA metabarcoding in these types of environments, then, it is important to distinguish "prey" DNA derived from metabarcoding results that is ingested from "prey" that may have been encountered in the shared environment. 

### Analyses outline

#### The data 

The data for these analyses consist of 19 spiders of the species *Heteropoda venatoria* collected on Palmyra Atoll in 2017. All spiders were starved in mesocosms for 12 hours and then fed a grasshopper (*Oxya japonica*). Spiders and grasshoppers were kept in mesocosms for 24 hours, after which all predators who had killed (and presumably ingested some or all of) the diet item were frozen at -20C. We surface sterilized 8 spider individuals by submerging them (after death in freezer) in 10% bleach for 2 minutes, and then washing them in deionized water for another 2 minutes. The remaining 11 spiders were not surface sterilized. Spiders were stored in 80% EtOH prior to DNA extraction. 

```{r experimental setup mesocosms, echo=FALSE, include=TRUE, out.width="100%", fig.align='center', fig.cap = "Figure 2: The mesocosm surface contamination experimental setup with sample sizes."}
knitr::include_graphics(here("Pictures", "mesocosm_experiment.png"))
```

After DNA extraction, PCR amplification, and sequencing on an Illumina MiSeq platform, sequences were quality filtered and denoised using two common denoising algorithms (UNOISE3 and DADA2). Both of these denoising algorithms take into account DNA sequence abundance and quality to combine sequences into Amplicon Sequence Variants (ASVs). Unlike traditional assignment methods (such as operational taxonomic units (OTUs)), which cluster sequences based on 97% sequence similarity, ASV denoising takes into account both sequence quality and abundance, and can detect differences and separate sequences that differ by one or very few nucleotides (ASVs are sometimes refered to as exact sequence variant (ESVs)). However, because UNOISE3 performed better than DADA2 in capturing prey ASV abundance and diversity (see other R Markdown), we chose to do all subsequent analyses with this pipeline only, (and will repeat on DADA2 to show that it doesn't change the outcome).

```{r ASV assignment figure, include=TRUE, echo=FALSE, out.width="100%", fig.align='center', fig.cap = "Figure 3: Denoising algorithms (including DADA2 and UNOISE3) consider sequence abundance and error rate in raw sequence datasets when removing sequences that have high error and low abundance and combining their reads with those of their most similar neighbor sequence with high abundance and low error (amplicon sequence variants or ASVs)."}
knitr::include_graphics(here("Pictures", "denoise_picture.png"))
```

Our final dataset for these mesocosm spiders includes ASVs assigned to known prey and the abundance of these ASVs in each sample. In addition, we have some mesocosm spiders (8) which we surface sterilized and some which we did not surface sterilize (11). *(Note: These numbers are different because of lower success in extracting DNA from surface sterilized mesocosm spiders.)*

#### The analyses

We will be looking at the whether surface sterilization alters our understanding of mesocosm feeding interactions in two ways:

1. Known prey detection per sample (*response: known prey ASV presence-absence*):

**Does surface sterilization alter our ability to detect known prey?**

  + Surface sterilization could increase the detection of known prey by removing surface contamination that may drown out actual prey during the PCR amplification and sequencing processes. 
  
  + Surface sterilization could decrease the detection of known prey by removing surface contamination coming from known prey encountered by the predator in the mesocosm environment either through shared surfaces or prey handling without ingestion.

 *glm(detection ~ surface sterilization, family = binomial)*
 
2. Known prey abundance per sample (*response: known prey read abundance of rarefied reads*):

**Does surface sterilization alter how much known prey we can recover from the metabarcoding process?**
    
  + Surface sterilization could increase the abundance of known prey, again by removing surface contamination that drowns out known diet. 
    
  + Surface sterilization could decrease the abundance of known prey, again by removing known prey DNA on the surface of predators encountered in the mesocosm environment.

 *glm(abundance ~ surface sterilization, family = poisson)*
  
#### The results of these analyses are summarised in the [summary table](#summary) along with some initial [ecological interpretations](#ecology) and [alternative hypothesis explorations](#alternative) .

### Set-up

#### Import ASV tables

First, we will import the data from both denoising algorithms (DADA2 and UNOISE3) and the metadata attached to each of the spiders in our dataset, including whether they were surface sterilized or not.

```{r meso what the dfs are, eval=FALSE}
u3_comm <- read.csv(here("data", "ASV_tables", "unoise_uc_zotu_tab.txt"), sep = '\t')
 
#sample metadata 
metadata <- read.csv(here("data", "Sample_Metadata.csv")) 
```


```{r meso import datasets and change column names}
u3_comm <- read.csv(here("data", "ASV_tables", "unoise_uc_zotu_tab.txt"), sep = '\t')

#rename X to ASV in all these tables.
u3_comm <- rename(u3_comm, "ASV" = "X.OTU.ID")

#rename all the sample names across dataframes for consistency
colnames(u3_comm) <- sapply(str_split(colnames(u3_comm), "S"), function(x){return(x[[1]])})
u3_comm <- rename(u3_comm, "ASV" = "A")

#sample metadata
metadata <- read.csv(here("data", "Sample_Metadata.csv"))
```

The ASV tables are matrices with ASVs as rows, and samples as columns:

```{r meso sample df heads}
u3_comm[1:10, 1:10]
```

And the metadata includes important data for each sample:

```{r meso metadata df head}
metadata
```

#### Assign taxonomies

For each of the ASVs in each of these tables, we used BLAST (with GenBank) and the BOLD ID Engine to assign taxonomies to these ASVs. We will import the dataframes of these taxonomic assignments for each algorithm and then subset the taxonomies that belong to the known prey item (here, either *Acrididae* or *Oxya* depending on the taxonomic assignment tool).

```{r meso example of taxonomy DBs, eval=FALSE}
#import both NCBI and BOLD taxonomies
u3_ncbi <- read.csv(here("6_taxonomic_assignment", "MEGAN", "unoise_unclean",
                            "unoise_uc_ncbi_taxa.csv"))
u3_bold <- read.csv(here("6_taxonomic_assignment", "BOLD", "usearch_uc", 
                            "unoise_uc_bold_taxa.csv"))
```


```{r meso assign taxonomies}
#import both NCBI and BOLD taxonomies
u3_ncbi <- read.csv(here("6_taxonomic_assignment", "MEGAN", "unoise_unclean",
                            "unoise_uc_ncbi_taxa.csv"))
u3_bold <- read.csv(here("6_taxonomic_assignment", "BOLD", "usearch_uc", 
                            "unoise_uc_bold_taxa.csv"))
u3_bold <- rename(u3_bold, "ASV" = "Query.ID")

#Sort by "predator", "prey", and "non-diet" before merging into one 
#dataframe for assigning to the abundance table!
#this following ifelse set will depend on the taxonomy of the set of samples
#and for this one, these are the categories that seemed to be appropriate based
#on the taxonomy list after looking at it
#sort into categories for bold:
u3_bold$taxonomy <- ifelse(
  u3_bold$ID == "Heteropoda venatoria", "predator",
  ifelse(u3_bold$Kingdom == "Fungi" | u3_bold$Class == "Mammalia", "non-diet", 
         ifelse(u3_bold$Phylum == "Arthropoda" | u3_bold$Class == "Reptilia" & u3_bold$ID != "Heteropoda venatoria", 
                "prey", "non-diet"
         )))

#sort into categories for ncbi:
u3_ncbi$taxonomy <- u3_ncbi$Category

#thus, the taxonomy variable has 4 levels: predator, prey, non-diet, non-hit (which
#we will assign below after merging with full ASV table)
#subset only the variables of interest from these two dataframes before merging
u3_bold_id <- u3_bold %>%
  dplyr::select(ASV, ID, Order, taxonomy)

u3_ncbi_id <- u3_ncbi %>%
  dplyr::select(ASV, ID, Order, taxonomy)

#join them together by ASV and taxonomy
u3_id_all <- u3_bold_id %>%  
  full_join(u3_ncbi_id, by = c("ASV", "taxonomy"))

u3_id_all <- u3_id_all %>%
  rename(ID_bold = ID.x,
         Order_bold = Order.x,
         ID_ncbi = ID.y,
         Order_ncbi = Order.y)

#this binds them all to the community matrix for analyses
u3_id_tab <- u3_comm %>%
  left_join(u3_id_all, by = "ASV")

#set NA taxonomies from join with "no hit" since they didn't match to any
#taxonomic assignments within our prey, predator, or non-prey categories
u3_id_tab$taxonomy <- replace_na(u3_id_tab$taxonomy, "no hit")

#write to a file for import into field analyses
u3_taxonomies <- u3_id_tab %>%
  dplyr::select(ASV, ID_bold, Order_bold, ID_ncbi, Order_ncbi, taxonomy)
write.csv(u3_taxonomies, here("data", "u3_uc_tax_ass.csv"))
```

Each taxonomic assignment includes identifiers from the Kingdom level down to Order, Family, Genus, or Species. Here are a few examples of taxonomies assigned to order and lower:

```{r meso ex taxonomy assignment}
u3_ncbi %>%
  dplyr::select(ASV, Order, Family, Genus, Species) %>%
  filter(ASV %in% c("ASV_1", "ASV_100", "ASV_105", "ASV_106", "ASV_154", 
                    "ASV_5", "ASV_6", "ASV_107"))
```

### Known prey ASVs: Analyses

We are first going to look at whether surface sterilization changes the detection of known prey ASVs in our dataset. Again, roughly half of the samples were surface sterilized and the other half were not surface sterilized. Furthermore, we ran two different bioinformatic algorithms on ALL samples. We will be running a generalized linear model with the full format:

glm(detection ~ surface_sterilization, family = binomial)

And then we will do model selection altering just the surface sterilization term, since this is the term we are interested in in the model. This full model is asking whether surface sterilization or denoising algorithm matter, and whether the relationship with them varies depending on the algorithm considered. 

#### Data set up: subset mesocosm spiders, rarefy samples, and subset known prey ASVs

Right now, this dataset includes all the spiders in our project, including those which we will analyze in Part 2 (field spiders) and all ASVs, including both our known prey ASVs and everything else detected in these samples, so we will need to subset the mesocosm spiders and the known prey ASVs from this dataset. After subsetting the mesocosm spiders and before subsetting the known prey ASVs and doing any analyses, however, we need to rarefy our samples to account for unequal sequencing depths that result from random sequencing preferences in the sequencing process on the Illumina MiSeq. To illustrate, if we consider the number of sequences in each sample in our UNOISE3 dataset, we can see that read abundance can vary a lot across samples:

```{r meso example sequence depth diffs setup}
lab <- as.character(metadata$sample[which(metadata$Source == "LAB")]) #vector of the sample names for lab samples

u3_commr <- u3_comm
rownames(u3_commr) <- u3_commr$ASV
u3_commr <- u3_commr %>%
  dplyr::select(all_of(lab))
```

```{r meso example seq depth diff}
hist(colSums(u3_commr))

colSums(u3_commr)
#max read abundance 
max(colSums(u3_commr))

#min read abundance
min(colSums(u3_commr))
```

This amount of variation in sequencing depth could lead to potentially misleading interpretations of any known prey detection or read abundance derived from this distribution. Thus, we will use the sample with the lowest total sequencing depth to rarefy our dataset so that we are comparing samples with similar sampling effort. 


```{r meso rarefy}
u3rare <- min(colSums(u3_commr))
u3rare #55205

set.seed(1)
u3_comm_rare <- as.data.frame(t(rrarefy(t(u3_commr), sample = u3rare)))
```

Now, all our samples have equal sampling depths, allowing for comparisons across samples:

```{r meso sampling depth example}
colSums(u3_comm_rare)
```

#### Subset just prey from this dataset

Once we've rarefied, we can subset only the prey DNA from this dataset, and start our analyses: 

```{r meso known prey read binding and selection}
u3_comm_rare$ASV <- rownames(u3_comm_rare)

u3_comm_rare %>%
  left_join(u3_id_all, by = "ASV") %>%
  filter(ID_ncbi == "Acrididae") %>%
  tally()

#this binds them all to the community matrix for analyses
u3_rare <- u3_comm_rare %>%
  left_join(u3_id_all, by = "ASV") %>% #combine with taxonomy DF
  filter(ID_ncbi == "Acrididae") %>% #select known diet only
  dplyr::select(-ID_bold, -ID_ncbi) %>% #remove taxonomy columns
  group_by(ASV) %>% 
  gather(sample, reads, HEV07:HEV29, factor_key = TRUE) %>% #make long while grouped by ASV
  group_by(sample) %>%
  summarise(reads = sum(reads)) #summarise total prey reads by sample 


all_rare <- u3_rare %>%
  left_join(metadata, by = "sample")
```

Now we're ready to do both our presence-absence AND abundance analyses for these mesocosm spiders!

#### Binomial GLM of prey detection

```{r meso set up for binomial glmm}
#create variable for surface sterilization
all_rare$surf_ster <- as.factor(ifelse(all_rare$Sterilized == "NS", 0, 1)) 

#create variable for presence-absence 
all_rare$Presence <- ifelse(all_rare$reads == 0, 0, 1)
```

Now we can run a set of models assessing whether surface sterilization is important for these samples. We will be comparing a model that includes surface sterilization to a model that does not include surface sterilization as a term (null model) 

```{r meso model for ASV presence}
ASV_ster_mod <- glmmTMB(Presence ~ Sterilized,
                    data = all_rare, 
                    family = binomial)

ASV_ster_null <- glmmTMB(Presence ~ 1,
                     data = all_rare,
                     family = binomial)

AICc(ASV_ster_mod, ASV_ster_null)
```

Based on these AIC values, the model with surface sterilization is the best fit for these data. 

And now we can look at some summaries of this model, which show that it's a good model, and that surface sterilization is a significant term in the model.

```{r meso model ASV summaries}
summary(ASV_ster_mod)
```

```{r meso model ASV fit}
simulationOutput_ASV <- simulateResiduals(fittedModel = ASV_ster_mod) 
ASV_fit <- plot(simulationOutput_ASV, asFactor=TRUE)
```

This is a fairly good model fit. We can further look at all the effects in this model by plotting the effects and then comparing the pairwise estimated marginal means of surface sterilization vs. no surface sterilization for each pipeine.

```{r meso model ASV pairwise}
plot(allEffects(ASV_ster_mod))
model.emm_ASV <- emmeans(ASV_ster_mod, "Sterilized")
pairs(model.emm_ASV)
```

What these results show is that surface sterilization marginally decreases known prey detection.  

```{r meso ASV visualization set up}
#let's visualize this then
ASVs_known <- all_rare %>%
  group_by(Sterilized) %>%
  summarise(total = n(), presence = sum(Presence), detection = presence/total)
```

```{r meso ASV detection visualization by sterilized}
my_y_title <- expression(paste(italic("O. japonica"), " ASV detection (%)"))

(a <- ggplot(ASVs_known, aes(x = Sterilized, y = detection*100)) +
  geom_bar(stat="identity", position= "dodge", fill = "#7B8D65", color = "black") +theme_bw() +
  labs(y = my_y_title) +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_blank()))
```

```{r asvs known summaries}
ASVs_known

all_rare %>%
  summarize(total = n(), present = sum(Presence), detection = present/total)
```  
  
#### Summary: Sterilization marginally decreased the detection of prey ASVs for both denoising algorithms. 

### Known prey reads: Analyses

Now, we are going to look at whether surface sterilization changes the abundance of known prey reads in our dataset. Again, we will be running a generalized linear model, now with a full format:

glm(abundance ~ surface_sterilization, family = poisson)

And then we will do model selection altering the surface sterilization term and comparing to a null model without this fixed effect. This full model is asking whether surface sterilization matters. 

For this section, we will be subsetting our data to *only* analyse the data for which prey DNA was detected. In this way, we are asking - when prey DNA is detected, does surface sterilization alter how much of it is detected. 

#### GLM for surface sterilization effects on read abundance

Now we can run a set of two models assessing whether surface sterilization is important for read abundances in these samples. We will be comparing a model with surface sterilization to one without it.

```{r meso sterilization and abundance models poisson}
all_rare_nz <- all_rare %>%
  filter(reads > 0)

read_ster_mod <- glmmTMB(reads ~ Sterilized,
                    data = all_rare_nz, 
                    family = poisson)

read_ster_null <- glmmTMB(reads ~ 1,
                     data = all_rare_nz,
                     family = poisson)

AICc(read_ster_mod, read_ster_null) #first model better, but let's check model assumptions before we move on
```

```{r meso ster abundance model fits poisson}
simulationOutput_knownprey <- simulateResiduals(fittedModel = read_ster_mod) 
knownpreyfit <- plot(simulationOutput_knownprey)
```

These models are not great, so we will fit with a negative binomial model instead:

```{r meso ster and abundance models NB}
read_ster_mod3 <- glmmTMB(reads ~ Sterilized,
                    data = all_rare_nz,
                    family = "nbinom2")

read_ster_null2 <- glmmTMB(reads ~ 1,
                     data = all_rare_nz,
                     family = "nbinom2")

AICc(read_ster_mod3, read_ster_null2) #the model with sterilization is still better, but marginally
```

```{r meso model fits abund and NB}
simulationOutput_knownprey <- simulateResiduals(fittedModel = read_ster_mod3) 
knownpreyfit <- plot(simulationOutput_knownprey) #these look okay
knownpreyzi <- testZeroInflation(simulationOutput_knownprey) #not zero inflated
knownpreyod <- testDispersion(simulationOutput_knownprey) 
#this is a lot better!
```

```{r meso residuals NB ster model}
plot(residuals(read_ster_mod3)) 
```


```{r meso reads model summary and marginal means}
summary(read_ster_mod3)

plot(allEffects(read_ster_mod3))

model.emm_reads <- emmeans(read_ster_mod3, "Sterilized")

pairs(model.emm_reads)
```

This model shows that sterilization increases prey reads. We can visualize the pattern here:

```{r meso visualize prey read abundance}
#let's visualize this then
my_y_title_2 <- expression(paste("Percent ", italic("O. japonica"), " reads per sample"))

b_1 <- ggplot(all_rare_nz, aes(x = Sterilized, y = (reads/u3rare)*100)) +
  geom_boxplot(fill = "#7B8D65") +
  theme_bw() + theme(legend.position = "none") +
  labs(x = "Sterilization treatment", y = my_y_title_2) +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized")) 

b_2 <- ggplot(all_rare_nz, aes(x = Sterilized, y = (reads/u3rare)*100)) +
  geom_boxplot(fill = "#7B8D65") +
  scale_y_log10() +
  theme_bw() + theme(legend.position = "none") +
  labs(x = "Sterilization treatment", y = my_y_title_2) +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())
```

```{r}
all_rare_nz %>%
  summarise(mean = mean(reads/u3rare), total =n(), sd = sd(reads/u3rare), se = sd/sqrt(total))

all_rare_nz %>%
  group_by(Sterilized) %>%
  summarise(mean = mean(reads/u3rare), total =n(), sd = sd(reads/u3rare), se = sd/sqrt(total))
```

It is important to note that this pattern is driven by one value and that the rest of the values show the opposite trend, highlighting the variability in read abundances in these kinds of datasets and an explanation of why presence-absence may be a better metric in this context. 

#### Summary Version 1: Sterilization marginally increased the amount of known prey in samples. 

### Caveat: an outlier driving pattern?

When looking at the distribution of read abundances in these samples, however, we noticed one thing, which is that one sample in the sterilized group has a read abundance an order of magnitude greater than any other sample:

```{r meso hist of read abundances}
hist(all_rare$reads)
```

And when we remove that sample, all samples have more equal read abundances in them:

```{r meso hist of read abundances minus outlier}
all_rare1 <- all_rare_nz %>%
  filter(sample != "HEV07")

hist(all_rare1$reads)
```

Without this outlier, let's reassess our model fits:

```{r meso sterilization and abundance models poisson remove outlier}
read_ster_mod4 <- glmmTMB(reads ~ Sterilized,
                    data = all_rare1,
                    family = "nbinom2")

read_ster_null3 <- glmmTMB(reads ~ 1,
                     data = all_rare1,
                     family = "nbinom2")

AICc(read_ster_mod4, read_ster_null3) #the model without sterilization is now better
```


```{r meso model fits read abundance sans outlier}
simulationOutput_knownprey <- simulateResiduals(fittedModel = read_ster_null3) 
knownpreyfit <- plot(simulationOutput_knownprey) 
knownpreyzi <- testZeroInflation(simulationOutput_knownprey) #not zero inflated
knownpreyod <- testDispersion(simulationOutput_knownprey) 
```


```{r meso summaries of nb reads model sans outlier}
summary(read_ster_null3)

plot(allEffects(read_ster_mod4))

model.emm_reads <- emmeans(read_ster_mod4, "Sterilized")

pairs(model.emm_reads)
```

There is no significant difference between sterilization treatments in the number of reads once the outlier is removed. 

#### Summary Version 2: Surface sterilization does not significantly alter known prey read abundance. 

```{r reads sum numbers}
all_rare1 %>%
  group_by(Sterilized) %>%
  summarise(mean = mean(reads/u3rare), n(), se = sd(reads/u3rare)/sqrt(`n()`))

all_rare1 %>%
  ungroup() %>%
  summarise(mean = mean(reads/u3rare), n(), se = sd(reads/u3rare)/sqrt(`n()`))
```


### Overall summary {#summary}

```{r meso summary table}
headers <- c("", "Outcome")
row1 <- c("Known prey ASV detection", "Surface sterilization marginally decreases detection of known prey")
row2 <- c("Known prey read abundance", "Surface sterilization *increases/does not change known prey read abundance")
row3 <- c("*swayed by one high value", "")

summary_table <- as.data.frame(rbind(headers, row1, row2, row3))
colnames(summary_table) <- summary_table$headers
knitr::kable(summary_table[2:4,], col.names = headers, row.names=FALSE)


```

### Biological/Ecological interpretations {#ecology}

**Known prey may be surface contamination from shared environment.** In a mesocosm environment, predators and prey may come into contact with each other in the mesocosm environment, either through shared surfaces or through the process of prey capture and handling. In our DNA metabarcoding data, some of the DNA that could be assigned as "ingested prey" may be contamination from shared space and surface contact in the mesocosm. While our mesocosms were simplified environments in which we could observe one predator individual interacting with one prey item, other mesocosm experiments can include multiple individuals or species of predators and prey that may or may not be large enough to observe interacting. In these contexts, confirming that prey detection is due to ingestion versus environmental contact is crucial for biological inference. The lack of detection in some surface sterilized individuals (5 or `r 5/8` of total) is also an interesting observation in our study, and may highlight that just because prey handling and mortality is observed, prey ingestion may not have occured. Alternatively, lack of detection represents limitations to the DNA metabarcoding approach to assigning prey items in predator diets. In any case, it seems that in contained environments, it is advisable to surface sterilize consumers, or at least to consider over-representation of prey ingestion if consumers are not surface sterilized prior to DNA extraction. 

```{r meso summary graphs again, out.width = '100%'}
#plot_grid(a, b)
```

#### Alternative significance {#alternative}

There is also an alternative explanation that may be that reduced detection of known prey could be a sign of DNA deterioration or degradation. While this is possible, because 1) prey DNA abundance does not decrease with surface sterilization, 2) other, less abundant, prey items do not decrease with surface sterilization, and 3) the field-collected spiders (spoiler) show no reduction in prey DNA with surface sterilization, this alternative hypothesis seems unlikely in this case. 


```{r meso predator DNA, include = FALSE}
#this binds them all to the community matrix for analyses
u3_pred <- u3_comm_rare %>%
  left_join(u3_id_all, by = "ASV") %>% #combine with taxonomy DF
  filter(taxonomy == "predator") %>% #select predator DNA only
  dplyr::select(-ID_bold, -ID_ncbi) %>% #remove taxonomy columns
  group_by(ASV) %>% 
  gather(sample, reads, HEV07:HEV29, factor_key = TRUE) %>% #make long while grouped by ASV
  group_by(sample) %>%
  summarise(reads = sum(reads)) %>% #summarise total prey reads by sample 
  left_join(metadata, by = "sample")


read_pred_u3 <- glmmTMB(reads ~ Sterilized,
                    data = u3_pred,
                 family = genpois)

read_pred_u3_null <- glmmTMB(reads ~ 1,
                    data = u3_pred,
                    family = genpois)

AICc(read_pred_u3,read_pred_u3_null) 

simulationOutput_pred <- simulateResiduals(fittedModel = read_pred_u3) 
predfit <- plot(simulationOutput_pred) 
predzi <- testZeroInflation(simulationOutput_pred) 
predod <- testDispersion(simulationOutput_pred)
plot(residuals(read_pred_u3))

hist(u3_pred$reads)

plot(allEffects(read_pred_u3))

model.emm_reads <- emmeans(read_pred_u3, "Sterilized")

pairs(model.emm_reads)

u3_pred %>%
  group_by(Sterilized) %>%
  summarise(mean = mean(reads), total = n(), se = sd(reads)/sqrt(total))

```

```{r meso pred graph, include = FALSE}
predators <- ggplot(u3_pred, aes(x = Sterilized, y = reads)) +
  geom_boxplot(fill = "#7B8D65") +theme_bw() +
  labs(x = "Surface sterilization treatment", y = "Predator DNA reads",
       title = "Predator DNA reads") +
    scale_y_sqrt() +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized")) + theme(legend.position = "none")
```


```{r meso other prey DNA, include = FALSE}
#this binds them all to the community matrix for analyses
u3_ot_prey <- u3_comm_rare %>%
  left_join(u3_id_all, by = "ASV") %>% #combine with taxonomy DF
  filter(taxonomy == "prey") %>% #select prey DNA only
  filter(!ID_ncbi == "Acrididae") %>% #don't choose known prey item
  dplyr::select(-ID_bold, -ID_ncbi) %>% #remove taxonomy columns
  group_by(ASV) %>% 
  gather(sample, reads, HEV07:HEV29, factor_key = TRUE) %>% #make long while grouped by ASV
  group_by(sample) %>%
  summarise(reads = sum(reads)) %>% #summarise total prey reads by sample 
  left_join(metadata, by = "sample") %>%
  mutate(pipeline = "u3")

#U3:
read_prey_u3 <- glmmTMB(reads ~ Sterilized,
                    data = u3_ot_prey,
                 family = genpois)

read_prey_u3_null <- glmmTMB(reads ~ 1,
                    data = u3_ot_prey,
                    family = genpois)

AICc(read_prey_u3,read_prey_u3_null) 

simulationOutput <- simulateResiduals(fittedModel = read_prey_u3_null) 
fit <- plot(simulationOutput) 
zi <- testZeroInflation(simulationOutput) 
od <- testDispersion(simulationOutput)
plot(residuals(read_prey_u3_null))


all_prey <- u3_ot_prey %>%
  group_by(Sterilized) %>%
  summarise(mean = mean(reads), total = n(), se = sd(reads)/sqrt(total))
```

```{r meso other prey graph, include=FALSE}
prey <- ggplot(u3_ot_prey, aes(x = Sterilized, y = reads)) +
  geom_boxplot(fill = "#7B8D65") +theme_bw() +
  labs(x = "Surface sterilization treatment", y = "Other prey DNA reads",
       title = "Prey DNA reads") +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized")) +  theme(legend.position = "none")
```

```{r meso pred prey DNA mesocosms, out.width = '100%'}
plot_grid(predators, prey, nrow=1)
```

#### [Back to the top](#top)

## Field Environment

### Field Environment Analyses

This code examines the diets of field-collected predators (the spider *Heteropoda venatoria*), specifically looking at the detection, diversity, and abundance of prey DNA and how surface sterilization of predator individuals might alter these measures of diet diversity. 

Specifically, we are exploring:

#### Does surface sterilization alter prey ASV detection or read abundance by removing surface contamination?

### Why it matters

Contained environments (both natural and artificial) are regularly used as model systems for understanding the effects of consumptive interactions on community structure (CITE mesocosms AND pools, or similar contained environments here). DNA metabarcoding offers a new way to confirm the effects of predator consumption in these environments, and, for example, separate consumption- vs. fear-based functional responses to community structure (Shmitz CITE), and maybe even get at numerical/biomass consumption (CITE the stuff from Mahon about intra-specific gene variation). In any of these environments, predators and prey may come into contact with each other in their environment through shared surfaces or substrates. When using DNA metabarcoding in these types of environments, then, it is important to distinguish "prey" DNA derived from metabarcoding results that is ingested from "prey" that may have been encountered in the shared environment. 

Many environments contain contaminant DNA on surfaces (CITE relic DNA or similar). Therefore, even in environments where consumers are not contained with prey, verifying that DNA attributed to "prey" in metabarcoding studies is ingested versus environmental contamination is key to making ecological inferences. Some environments may be more or less prone to environmental contamination, and so the decision to surface sterilize may depend on the environment in which consumers are collected. Again, just as in contained environments, contamination on the outside of prey may alter or inflate prey detection, abundance, diversity, or species composition. Determining whether surface contamination may alter ecological inference is key for any diet metabarcoding study for which entire consumers are used. 

### Analyses outline

#### The data

The data for these analyses consist of 37 individuals of the spider species *Heteropoda venatoria* collected on Palmyra Atoll in 2015. All spiders were collected in natural habitats on the atoll in individual collection chambers. All spiders were frozen at -20C immediately after capture and then transferred to vials in a -80C freezer until DNA extraction in 2019. We surface sterilized 18 individuals by submerging them in 10% bleach for 2 minutes, and then washing them in deionized water for another 2 minutes. The remaining 19 spider individuals were not surface sterilized.

```{r experimental setup, echo=FALSE, include=TRUE, out.width="100%", fig.align='center', fig.cap = "Figure 3: The field-collected surface contamination experimental setup with sample sizes."}
knitr::include_graphics(here("Pictures", "field_experiment.png"))
```

After DNA extraction, PCR amplification, and sequencing on an Illumina MiSeq platform, sequences were quality filtered and denoised using two common denoising algorithms (UNOISE3 and DADA2). Both of these denoising algorithms take into account DNA sequence abundance and quality to combine sequences into Amplicon Sequence Variants (ASVs). Unlike traditional assignment methods (such as operational taxonomic units (OTUs)), which cluster sequences based on 97% sequence similarity, ASV denoising takes into account both sequence quality and abundance, and can detect differences and separate sequences that differ by one or very few nucleotides (ASVs are sometimes refered to as exact sequence variant (ESVs)) (Figure 2). We only did final analyses on UNOISE3 since it performed better than DADA2 in important ways. 

Our final dataset for these field-collected spiders includes ASVs assigned to prey taxonomies and the abundance of these ASVs in each sample. In addition, we have some mesocosm spiders (18) which we surface sterilized and some which we did not surface sterilize (19). 

#### The analyses

We will be looking at the whether surface sterilization alters our understanding of predator-prey feeding interactions


1. Big picture patterns in diversity:

**Does surface sterilization alter the abundance and diversity of prey DNA?**

  + Prey DNA detection by sample 
  
      **glm(Presence ~ surface_sterilization, family = binomial)**
    
  + Total species richness of all prey by sample
  
      **glm(SR ~ surface_sterilization, family = poisson)**
      
  + Prey ASV read abundance by sample
  
      **glm(reads ~ surface_sterilization, family = poisson)**
 
2. Zooming in to changes in community composition:

**Does surface sterilization alter the community composition of prey DNA?**

  + Jaccard presence-absence community analysis of rarefied prey DNA
  
      **glmer/glmmTMB(presence ~ surface_sterilization + (1 + surface_sterilization | prey_species), family = binomial)** 
      
  + Bray-Curtis abundance-based community dissimilarlity of rarefied prey DNA
  
    **adonis(community ~ surface_sterilization, dist = "bray")**
      
#### The results of these analyses are summarized in the [summary table](#summaryfield) along with some initial [ecological interpretations](#ecologyfield).

### Set-up

We do not need to add the taxonomies to our dataframes, since this was already done through the mesocosm spider data set up. In this case, we need to subset field spiders, rarefy these samples basd on lowest read abundance, subset just the prey DNA ASVs from this, and then combine prey ASVs based on duplicate taxonomies.

#### Data set up: subset field spiders and rarefy

**Subset field spiders, rarefy samples, combine duplicate prey DNA taxonomies, and subset ASVs to include only those which are prey ASVs**

Right now, this dataset includes all the spiders in our project, including those which we analyzed in Part 1 (mesocosm spiders) and all ASVs, including both prey ASVs and everything else detected in these samples, so we will need to subset the field spiders and then prey ASVs from this dataset. We will also need to combine duplicate taxonomic assignments for these prey ASVs prior to any analysis of species richness or community composition. 

After subsetting the field spiders and before subsetting and combining the prey ASVs and doing any analyses, however, we need to rarefy our samples to account for unequal sequencing depths that result from random sequencing preferences in the sequencing process on the Illumina MiSeq. To illustrate, if we consider the number of sequences in each sample in our UNOISE3 dataset, we can see that read abundance can vary a lot across samples:

```{r field example sequence depth diffs setup}
field <- as.character(metadata$sample[which(metadata$Source == "FIELD")]) #vector of the sample names for field samples

u3_commf <- u3_comm
rownames(u3_commf) <- u3_commf$ASV
u3_commf <- u3_commf %>%
  dplyr::select(all_of(field))
```

```{r field example seq depth diff}
hist(colSums(u3_commf))

#max read abundance 
max(colSums(u3_commf))
#256204

#min read abundance
min(colSums(u3_commf))
#16004
```

This amount of variation in sequencing depth could lead to potentially misleading interpretations of any known prey detection or read abundance derived from this distribution. Thus, we will use the sample with the lowest total sequencing depth to rarefy our dataset so that we are comparing samples with similar sampling effort. 


```{r field rarefy}
u3raref <- min(colSums(u3_commf))
u3raref #16004

set.seed(1)
u3_comm_rare_f <- as.data.frame(t(rrarefy(t(u3_commf), sample = u3raref)))
```

Now, all our samples have equal sampling depths, allowing for comparisons across samples:

```{r field sampling depth example}
colSums(u3_comm_rare_f)
```

Now we're ready to do both our presence-absence AND abundance analyses for these field-collected spiders!

#### Data set up: subset just prey DNA from this dataset and then combine duplicate taxonomic assignments

```{r field subset just prey from dataset}
#now subset the tables with taxonomic assignments for the columns
#that match these samples
#d2_id_all and u3_id_all combined with the rarefied u3_comm_rare_f above
u3_comm_rare_f$ASV <- rownames(u3_comm_rare_f)
u3_fld <- u3_comm_rare_f %>% 
  left_join(u3_id_all, by = "ASV") %>%
  dplyr::select(ASV, all_of(field), ID_bold, ID_ncbi, taxonomy) %>%
  filter(taxonomy == "prey") #42 prey ASVs
```

```{r field combine duplicate taxonomies}
#(first, combine uniques) ##
#use d2_fld and u3_fld with uniques instructions below
#UNOISE3
#make the ID columns characters for the ifelse statements below
u3_fld$ID_bold <- as.character(u3_fld$ID_bold)
u3_fld$ID_ncbi <- as.character(u3_fld$ID_ncbi)

#ifelse statement that assigns each ASV to a specific taxonomy
#I want the BOLD ids to be the last option because these are to
#species in many cases where NCBI/Genbank blasted to order, family, or genus
#therefore, I'm going to say that if there is no BOLD id, give it the NCBI
#ID, otherwise, give it the BOLD ID.
IDs_u3 <- ifelse(is.na(u3_fld$ID_bold), u3_fld$ID_ncbi, 
                                               u3_fld$ID_bold)
IDs_u3

#make this a dataframe so we can attach it back to the community
#matrix for analyses at the sample level and later Jaccard dissimiliarity
IDs_u3 <- as.data.frame(IDs_u3)
IDs_u3$ASV <- u3_fld$ASV
IDs_u3 <- rename(IDs_u3, "unique_ID" = "IDs_u3")
unique_ID_u3 <- as.data.frame(unique(IDs_u3[c("unique_ID")]))
unique_ID_u3$sp_number <- seq.int(nrow(unique_ID_u3))

comm_u3 <- IDs_u3 %>%
  left_join(unique_ID_u3, by = "unique_ID") %>%
  left_join(u3_fld, by = "ASV") %>%
  gather(sample, reads, HEV65:HEV100) %>%
  left_join(metadata, by = "sample")
```

#### Data set up: summarize detection, richness, and abundance for analyses

Now, create summarized dataframes that include detection, richness, and abundance of prey reads in each sample.

```{r field detection richness and abundance}
#species richness of prey in each pipeline
all_rich <- comm_u3 %>%
  group_by(sample, sp_number, Sterilized) %>%
  summarize(reads = sum(reads)) %>% #this summarizes reads by each sp in each sample
  ungroup() %>%
  group_by(sample, Sterilized) %>% #then if we group by just sample, can count number of >0
  summarize(SR = sum(reads > 0, na.rm=TRUE)) %>% #count number of species >0
  mutate(pipeline = "u3") #add pipeline column to bind later

all_abund <- comm_u3 %>%
  group_by(sample, Sterilized) %>% 
  summarize(abund = sum(reads)) %>% #this counts total abundance of prey reads in a sample
  mutate(pipeline = "u3", presence = ifelse(abund > 0, 1, 0)) #this says wheter prey detected
```

Now let's look at our broad scale analyses, including prey detection, prey richness, and prey abundance.

### Big picture patterns in diversity:

We are first going to look at whether surface sterilization changes the detection of prey ASVs in our dataset. Again, roughly half of the samples were surface sterilized and the other half were not surface sterilized. We will be looking at prey detection, prey richness, and prey read abundance *per sample* and how sterilization may alter these metrics of prey diversity.

  1. Prey detection by sample 
  
      **glm(Presence ~ surface_sterilization, family = binomial)**
    
  2. Total species richness of all prey by sample
  
      **glm(SR ~ surface_sterilization, family = poisson)**
  
  For models, we we will do model selection comparing a model with surface sterilization as a term to one that does not include surface sterilization as a term.
  
  3. Prey ASV read abundance by sample
  
      **glm(reads ~ surface_sterilization, family = poisson)**

#### 1. Prey DNA presence in samples

```{r presence models}
#using all_abund and binomial models
#create variable for surface sterilization
pres_ster_mod <- glmmTMB(presence ~ Sterilized,
                    data = all_abund, 
                    family = binomial)

pres_ster_null <- glmmTMB(presence ~ 1,
                     data = all_abund,
                     family = binomial)

AICc(pres_ster_mod, pres_ster_null)
```

The null model looks like the best fit based on AICc values. 

```{r field model presence summaries}
summary(pres_ster_null)
binned_residuals(pres_ster_null)
```

```{r field model presence fit}
simulationOutput_pres <- simulateResiduals(fittedModel = pres_ster_null) 
pres_fit <- plot(simulationOutput_pres, asFactor=TRUE)
```

For prey presence, the null model which *did not* include surface sterilization is the best fit model.

```{r field visualize presence}
#let's visualize this then
prey_presence <- all_abund %>%
  group_by(Sterilized) %>%
  summarise(total = n(), presence = sum(presence), detection = presence/total)

(c <- ggplot(prey_presence, aes(x = Sterilized, y = detection)) +
  geom_bar(stat="identity", position= "dodge", fill = "#F29979", color = "black") +theme_bw() +
  labs(x = "Surface sterilization treatment", y = "Prey ASV detection (%)") +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized")) +
  theme(legend.position = "none"))
```


```{r field detection average}
all_abund %>%
  ungroup() %>%
  summarise(total = n(), presence = sum(presence), detection = presence/total)

all_abund %>%
  group_by(Sterilized) %>%
  summarise(total = n(), presence = sum(presence), detection = presence/total)

all_abund %>%
  dplyr::select(sample, presence) %>%
  rename("Presence" = "presence") %>%
  bind_rows(all_rare) %>%
  dplyr::select(sample, Presence) %>%
  ungroup() %>%
  summarise(total = n(), present = sum(Presence), detection = present/total)
```

#### Summary: Surface sterilization does not significantly alter prey DNA detection, with detection of prey DNA in roughly 75% of samples.

#### 2. Prey DNA richness 

```{r field richness models}
rich_ster_mod <- glmmTMB(SR ~ Sterilized,
                    data = all_rich, 
                    family = poisson)

rich_ster_null <- glmmTMB(SR ~ 1,
                    data = all_rich, 
                    family = poisson)

AICc(rich_ster_mod, rich_ster_null)
```

Model without surface sterilization is better fit.

```{r field richness model fit}
plot(residuals(rich_ster_null))

simulationOutput_rich <- simulateResiduals(fittedModel = rich_ster_null) 
rich_fit <- plot(simulationOutput_rich, asFactor=TRUE)
SRzi <- testZeroInflation(simulationOutput_rich) 
SRod <- testDispersion(simulationOutput_rich) 
```

The best model is the null model not including surface sterilization.

```{r field visualze richness}
(d <- ggplot(all_rich, aes(x = Sterilized, y = SR)) +
  geom_boxplot(fill = "#F29979") + theme_bw() +
  labs(x = "Surface sterilization treatment", y = "Prey richness") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1)) + 
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized")))
```


```{r field richness average}
all_rich %>%
  ungroup() %>%
  summarize(mean = mean(SR), min = min(SR), max = max(SR), total= n(), sd = sd(SR), se = sd/total)
```

#### Summary: Surface sterilization does not significantly alter prey DNA richness across samples. 

#### 3. Prey DNA abundance

```{r field prey abund models}
all_abund_nz <- all_abund %>%
  filter(abund > 0)

abund_ster_mod <- glmmTMB(abund ~ Sterilized,
                    data = all_abund_nz, 
                    family = poisson)

abund_ster_null <- glmmTMB(abund ~ 1,
                    data = all_abund_nz, 
                    family = poisson)

AICc(abund_ster_mod, abund_ster_null)
```

Based on these models, the full model with sterilization is the best fit, but let's see if it is a good model. 

```{r field abund model fit}
plot(residuals(abund_ster_mod))

simulationOutput_abund <- simulateResiduals(fittedModel = abund_ster_mod) 
abund_fit <- plot(simulationOutput_abund, asFactor=TRUE) #ver BAD
abundod <- testDispersion(simulationOutput_abund) #eek
```

Based on these diagnostics, we need to try to fit a NB distribution, since these data are very overdispersed.

```{r field nb prey abund models}
abund_ster_mod3 <- glmmTMB(abund ~ Sterilized,
                    data = all_abund_nz,
                    family = "nbinom2")

abund_ster_null2 <- glmmTMB(abund ~ 1,
                    data = all_abund_nz,
                    family = "nbinom2")

AICc(abund_ster_mod3, abund_ster_null2)
```

Based on these models, now, the null model is the best model fit, which does not include surface sterilization as a predictor of prey read abundance. Let's check model fits again.

```{r field nb abund model fit}
plot(residuals(abund_ster_null2))

simulationOutput_abund2 <- simulateResiduals(fittedModel = abund_ster_null2) 
abund_fit2 <- plot(simulationOutput_abund2, asFactor=TRUE) #sort of better?
abundod2 <- testDispersion(simulationOutput_abund2) #better
summary(abund_ster_null2)

plot(allEffects(abund_ster_mod3))
emm.means.nz2 <- emmeans(abund_ster_mod3, "Sterilized")
pairs(emm.means.nz2)
```


```{r field visualize abundance}
e_1 <- ggplot(all_abund_nz, aes(x = Sterilized, y = (abund/u3raref)*100)) +
  geom_boxplot(fill = "#F29979") +
  labs(x = "Surface sterilization treatment", y = "Percent prey reads per sample") +
  theme_bw() + theme(legend.position = "none") +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized"))

e_2 <- ggplot(all_abund_nz, aes(x = Sterilized, y = (abund/u3raref)*100)) +
  geom_boxplot(fill = "#F29979") +
  scale_y_log10()+
  labs(x = "Surface sterilization treatment", y = "Percent prey reads per sample") +
  theme_bw() + theme(legend.position = "none") +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized"))
```


```{r field abund average}
all_abund_nz %>%
  ungroup() %>%
  summarize(mean = mean(abund/u3raref), min = min(abund/u3raref), max = max(abund/u3raref), total = n(), sd = sd(abund/u3raref), se =sd/sqrt(total))
```

#### Aside: predator read abundance

```{r}
#this binds them all to the community matrix for analyses
u3_pred_f <- u3_comm_rare_f %>%
  left_join(u3_id_all, by = "ASV") %>% #combine with taxonomy DF
  filter(taxonomy == "predator") %>% #select predator DNA only
  dplyr::select(-ID_bold, -ID_ncbi) %>% #remove taxonomy columns
  group_by(ASV) %>% 
  gather(sample, reads, HEV65:HEV100, factor_key = TRUE) %>% #make long while grouped by ASV
  group_by(sample) %>%
  summarise(reads = sum(reads)) %>% #summarise total prey reads by sample 
  left_join(metadata, by = "sample")

ggplot(u3_pred_f, aes(x = Sterilized, y = reads)) +
  geom_boxplot(fill = "#7B8D65") +theme_bw() +
  labs(x = "Surface sterilization treatment", y = "Predator DNA reads",
       title = "Predator DNA reads") +
    scale_y_sqrt() +
  scale_x_discrete(labels=c("NS" = "Not S. Sterilized", "SS" = "Surface Sterilized")) + theme(legend.position = "none")
```

#### Summary: Surface sterilization does not significantly alter prey DNA abundance across samples.

### Zooming in; changes in community composition:

We are now going to look at finer-scale patterns of community composition in our samples. Specifically, we are going to explore whether surface sterilization changes the community composition of our samples (looking at the rarefied data, since this is a measure of beta diversity). Again, roughly half of the samples were surface sterilized and the other half were not surface sterilized. Furthermore, we ran two different bioinformatic algorithms on ALL samples. We will be looking at prey community composition *per sample* and how sterilization may alter community composition.
      
  1. Jaccard presence-absence community analysis of rarefied prey DNA
  
      **glmer/glmmTMB(presence ~ surface_sterilization + (1 + surface_sterilization | prey_species), family = binomial)** 
  
  2. Bray-Curtis abundance-based dissimilarity of rarefied prey DNA
  
      **adonis(community ~ surface_sterilization, dist = "bray")**
  
  For Jaccard models, we we will do model selection altering just the surface sterilization term, since this is the term we are interested in in the model. This full model is asking whether surface sterilization or denoising algorithm matter for changing community composition, and whether the relationship with them varies depending on the algorithm considered. Additionally, the random slopes term for surface sterilization is allowing the by-species slope of the relationship of surface steriliation to vary, which means that some species can be *more* abundant while others can be *less* abundant when samples were surface sterilized.
  
#### Jaccard and Bray-Curtis Abundance-based dissimilarity

```{r field bray dissimilarity}
diss_u3 <- u3_comm_rare_f
rownames(diss_u3) <- u3_comm_rare_f$ASV
diss_u3 <- diss_u3 %>%
  dplyr::select(-ASV) 

dissm_u3 <- as.data.frame(t(diss_u3))
dissm_u3 <- dissm_u3[which(colSums(dissm_u3) > 0) ] #176 to 135
dissm_u3 <- dissm_u3[which(rowSums(dissm_u3) > 0), ] #37, none removed
dissm_u3 <- as.matrix(dissm_u3)
```


```{r field adonis abundance}
meta_field <- metadata %>%
  filter(Source == "FIELD")

adonis(dissm_u3 ~ Sterilized, data = meta_field, dist = "jaccard", binary = TRUE)

adonis(dissm_u3 ~ Sterilized, data = meta_field, dist = "bray")
```


```{r field rarefied jaccard, include = TRUE}
rare_jacc_u3 <- comm_u3 %>%
  dplyr::select(unique_ID, sample, reads, Sterilized) %>% #select factors of interest
  ungroup() %>% #ungroup any grouping variables
  group_by(sample, unique_ID, Sterilized) %>% #regroup by sample, taxonomiy, and sterilizaion
  summarize(reads = sum(reads)) %>% #summarize the number of reads of each unique taxonomy acros samples
  mutate(pipeline = "u3", presence = ifelse(reads > 0, 1, 0)) #add pipeline so we could combine with each other, and then also create a presence column
 
#replace Oxya with Acrididae to talk to the Dada2 dataframe below
rare_jacc_u3$unique_ID <- as.character(rare_jacc_u3$unique_ID)
rare_jacc_u3$unique_ID <- ifelse(rare_jacc_u3$unique_ID == "Oxya", "Acrididae", rare_jacc_u3$unique_ID)


rare_jacc <- rare_jacc_u3 %>%
  ungroup() %>%
  mutate(unique_ID = as.factor(unique_ID))

#for Ruby's data
#diet <- rare_jacc %>%
#  rename(abundance = reads, Extract_ID = sample, diet_ID = unique_ID) %>%
#  dplyr::select(-Sterilized, -pipeline) %>%
 # filter(abundance > 0)
#write.csv(diet, "DNA_diet.csv")

levels(rare_jacc$unique_ID)
```

```{r field remove zeros, include = FALSE}
present_rare <- rare_jacc %>%
  group_by(unique_ID) %>%
  tally(reads > 0) %>%
  filter(n > 0) #27 of 32 uniques have non-zero values across all samples in rarefied dataset
```


```{r jaccard at ID level, include=TRUE}
rare_jacc2 <- present_rare %>%
  left_join(rare_jacc, by = "unique_ID")
  
jacc_model1 <- glmmTMB(presence ~ Sterilized + (1+Sterilized|unique_ID), 
                     data = rare_jacc2,
                     family = "binomial")

jacc_model_null <- glmmTMB(presence ~ 1 + (1|unique_ID), 
                     data = rare_jacc2,
                     family = "binomial")

AICc(jacc_model1, jacc_model_null)
```


```{r summary jaccard rare}
summary(jacc_model_null)
binned_residuals(jacc_model_null)
```

```{r model jaccard fit}
simulationOutput_jacc <- simulateResiduals(fittedModel = jacc_model_null)
jacc_fit <- plot(simulationOutput_jacc, asFactor=TRUE)
```

```{r effects plot of ID, include=TRUE}
jacc_graph1_u3 <- rare_jacc2 %>%
  filter(pipeline == "u3") %>%
  filter(unique_ID != "Acrididae") %>%
  group_by(unique_ID, Sterilized) %>%
  summarize(mean = mean(presence), sd = sd(presence)) %>%
  ungroup() %>%
  group_by(unique_ID) %>%
  pivot_wider(names_from = c(Sterilized),
              values_from = c(mean, sd)) %>%
  mutate(se_NS = sd_NS/sqrt(19), se_SS = sd_SS/sqrt(18))

es_ID_u3 <- esc_mean_se(grp1m = jacc_graph1_u3$mean_NS, grp1se = jacc_graph1_u3$se_NS, grp1n = 19,
  grp2m = jacc_graph1_u3$mean_SS, grp2se = jacc_graph1_u3$se_SS, grp2n = 18, es.type = "g")

IDs <- as.character(jacc_graph1_u3$unique_ID)
Hedges_g <- es_ID_u3$es
Lower_CI <- es_ID_u3$ci.lo
Upper_CI <- es_ID_u3$ci.hi

effects <- as.data.frame(cbind(IDs, Hedges_g, Lower_CI, Upper_CI))

effects$Hedges_g <- as.numeric(as.character(effects$Hedges_g))
effects$Lower_CI <- as.numeric(as.character(effects$Lower_CI))
effects$Upper_CI <- as.numeric(as.character(effects$Upper_CI))

u3_effect_graph <- ggplot(effects, aes(x = IDs, y = Hedges_g)) +
  geom_point() +theme_bw() +geom_hline(yintercept = 0) +
  geom_pointrange(aes(ymin = Lower_CI, ymax = Upper_CI)) +
  labs(title = "UNOISE3 effect size of average presence") +
  theme(axis.text.x = element_text(angle=90, hjust = 1)) + coord_flip()

u3_effect_graph
```


```{r field combine to order for vis}
u3_comm_ord <- u3_comm_rare_f %>%
  left_join(u3_id_all, by = "ASV") %>%
  filter(taxonomy == "prey") %>%
  dplyr::select(-taxonomy)

#UNOISE3
#make the ID columns characters for the ifelse statements below
u3_comm_ord$Order_ncbi <- as.character(u3_comm_ord$Order_ncbi)
u3_comm_ord$ID_ncbi <- as.character(u3_comm_ord$ID_ncbi)

#ifelse statements prioritizing NCBI since there are more.
Orders_u3 <- ifelse(u3_comm_ord$Order_ncbi == "", u3_comm_ord$ID_ncbi,
                    u3_comm_ord$Order_ncbi)
#Orders_u3

#make this a dataframe so we can attach it back to the community
#matrix for analyses at the sample level and later Jaccard dissimiliarity
Orders_u3 <- as.data.frame(Orders_u3)
Orders_u3$ASV <- u3_comm_ord$ASV
Orders_u3 <- rename(Orders_u3, "unique_order" = "Orders_u3")
unique_Orders_u3 <- as.data.frame(unique(Orders_u3[c("unique_order")]))
unique_Orders_u3$sp_number <- seq.int(nrow(unique_Orders_u3))

ords_comm_u3 <- Orders_u3 %>%
  left_join(unique_Orders_u3, by = "unique_order") %>%
  left_join(u3_comm_ord, by = "ASV") %>%
  gather(sample, reads, HEV65:HEV100) %>%
  left_join(metadata, by = "sample") %>%
  ungroup() %>%
  group_by(sample, unique_order, Sterilized) %>%
  summarize(reads = sum(reads)) %>%
  mutate(pipeline = "u3", presence = ifelse(reads > 0, 1, 0)) %>%
  group_by(unique_order) %>%
  filter(sum(presence) > 0) #from 518 to 407, removed 3 orders?


orders_comm <- ords_comm_u3 %>%
    ungroup() %>%
  mutate(unique_order = as.factor(unique_order), surf_ster=ifelse(Sterilized == "NS", 0, 1)) 

levels(orders_comm$unique_order)
```


```{r field dissimilarity heat map}
heat_map <- orders_comm %>%
  group_by(unique_order, Sterilized) %>%
  summarise(reads = mean(reads)) %>%
  mutate(presence = ifelse(reads > 0, 1, 0))

heat_map_nz <- heat_map %>%
  filter(reads > 0)
quantile(heat_map_nz$reads)

heat_map <- heat_map %>%
  ungroup() %>%
  dplyr::select(Sterilized, unique_order, reads) %>%
  pivot_wider(names_from = Sterilized, values_from = reads) %>%
  arrange((NS + SS), (NS)) %>%
  mutate(unique_order=factor(unique_order, levels=unique_order)) %>%
  gather(Sterilized, reads, NS:SS) 


heat_map$quantiles <- ifelse(heat_map$reads == 0, 0,
                             ifelse(heat_map$reads > 0 & heat_map$reads <= 0.1052632, 1, 
                                    ifelse(heat_map$reads > 0.1052632 & heat_map$reads <= 0.7456140, 2,
                                           ifelse(heat_map$reads > 0.7456140 & heat_map$reads <= 2.8128655, 3,
                                                  ifelse(heat_map$reads > 2.8128655 & heat_map$reads <= 21.5614035, 4, 5)))))

heat_map$quantiles <- as.factor(heat_map$quantiles)  

pal <- c(
  '0' = "#FFFFFF",
  '1' = "#F27D72", 
  '2' = "#D26F67", 
  '3' = "#B2615C",
  '4' = "#925451",
  '5' = "#734646"
)

pal2 <- c(
  '0' = "#FFFFFF",
  '1' = "#F29979", 
  '2' = "#D2846C", 
  '3' = "#B26F5F",
  '4' = "#925B53",
  '5' = "#734646"
)

heat_map_graph <- ggplot(heat_map, aes(x = Sterilized, y = unique_order, fill=quantiles, height = 0.95, width = 0.95)) +
  geom_tile() + 
  coord_equal() +
  labs(x = "Surface sterilization treatment", y = "Diet group") +
  scale_x_discrete(labels=c("NS" = "Not Sterilized", "SS" = "Surface Sterilized")) +
    scale_fill_manual(name = "Mean read abundance\n(divided by quantiles)",
    values = pal2,
    limits = names(pal2),
    labels = c("0", "< 0.1", "< 0.7", "< 2.8", "< 21.6", "< 219.5")) + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

```

```{r field dissimiarlity graph}
heat_map_graph
```

####Summary: Surface sterilization does not change diet community composition based on  presence-absence or abundance.

### Overall summary {#summaryfield}

```{r field summary table}
headers2 <- c("", "Outcome")
row1_2 <- c("Prey ASV detection", "Surface sterilization does not alter detection of prey")
row2_2 <- c("Prey ASV richness", "Surface sterilization does not alter prey richness")
row3_2 <- c("Prey ASV abundance", "Surface sterilization does not alter prey read abundance")
row4_2 <- c("Community composition", "Surface sterilization does not alter prey community composition*")
row5_2 <- c("*this includes both presence-absence and abundance-based composition", "")

summary_table_2 <- as.data.frame(rbind(headers2, row1_2, row2_2, row3_2, row4_2, row5_2))
colnames(summary_table_2) <- summary_table$headers2
knitr::kable(summary_table_2[2:6,], col.names = headers2, row.names=FALSE)
```

### Biological/Ecological interpretations {#ecologyfield}

**For this consumer in its natural environment, surface contamination does not alter ecologically important diet measures.** In un-contained environments, predators come into contact with surfaces that may be shared with potential prey items. In this setting, it seems that this surface contamination does not significantly alter the detection, abundance, or diversity (both richness or species composition) of DNA that can be attributed to "prey". Furthermore in this environment, we can see that we don't always detect prey DNA in predators - this could be due to limitations to DNA metabarcoding methods, or could be due to natural rates of consumption and starving in predator populations. The degree to which predators are "starving" or "fasting" in natural environments is an important ecological pattern in itself (CITE), and so this lack of detection across species could be useful for ecological inference. In this environment, while surface sterilization is not necessary, we see that having a conservative approach and choosing to surface sterilize consumers is also not a bad appraoch, as we did not observe loss in prey DNA when consumers were surface sterilized. 
                                 
```{r field summary graphs, out.width = '100%'}
#plot_grid(c, e, nrow = 1)

#plot_grid(d, heat_map_graph, align = "h", rel_widths = c(1,2))
```

#### [Back to the top](#top)

## Overall Summary 

### Overall:

The choice to surface sterilize consumers prior to DNA metabarcoding for diet analyses depends on environmental context, being apparently more important in contained environments where consumers and prey share substrates and surfaces more often. However, in all cases, choosing to be conservative and surface sterilize will not remove real prey items from analyses.  

### Mesocosm summary

**Known prey may be surface contamination from shared environment.** In a mesocosm environment, predators and prey may come into contact with each other in the mesocosm environment, either through shared surfaces or through the process of prey capture and handling. In our DNA metabarcoding data, some of the DNA that could be assigned as "ingested prey" may be contamination from shared space and surface contact in the mesocosm. While our mesocosms were simplified environments in which we could observe one predator individual interacting with one prey item, other mesocosm experiments can include multiple individuals or species of predators and prey that may or may not be large enough to observe interacting. In these contexts, confirming that prey detection is due to ingestion versus environmental contact is crucial for biological inference. The lack of detection in some surface sterilized individuals (5 or `r 5/8` of total) is also an interesting observation in our study, and may highlight that just because prey handling and mortality is observed, prey ingestion may not have occured. Alternatively, lack of detection represents limitations to the DNA metabarcoding approach to assigning prey items in predator diets. In any case, it seems that in contained environments, it is advisable to surface sterilize consumers, or at least to consider over-representation of prey ingestion if consumers are not surface sterilized prior to DNA extraction. 

```{r meso summary graphs again again, out.width = '100%'}
#plot_grid(a, b)
```

### Field summary

**For this consumer in its natural environment, surface contamination does not alter ecologically important diet measures.** In un-contained environments, predators come into contact with surfaces that may be shared with potential prey items. In this setting, it seems that this surface contamination does not significantly alter the detection, abundance, or diversity (both richness or species composition) of DNA that can be attributed to "prey". Furthermore in this environment, we can see that we don't always detect prey DNA in predators - this could be due to limitations to DNA metabarcoding methods, or could be due to natural rates of consumption and starving in predator populations. The degree to which predators are "starving" or "fasting" in natural environments is an important ecological pattern in itself (CITE), and so this lack of detection across species could be useful for ecological inference. In this environment, while surface sterilization is not necessary, we see that having a conservative approach and choosing to surface sterilize consumers is also not a bad appraoch, as we did not observe loss in prey DNA when consumers were surface sterilized. 

```{r field graphs again, out.width = '100%'}
#plot_grid(c, e, nrow = 1)

#plot_grid(d, heat_map_graph, align = "h", rel_widths = c(1,2))
```

### Future directions

Like any experiment, this project brings up important next directions in validating diet analyses with DNA metabarcoding. While we uncovered that surface contamination may be an issue with diet inference, depending on environmental context, other questions remain that are worth exploring, including:

  + Do you detect presumed prey if consumers and prey come into contact with each other, but ingestion has not occured? (Approach: consumers and prey contact on outer surfaces, surface sterilize some and not others.)

  + Does metabarcoding approach alter prey detection? (Approach: validate that consumers eat prey, rather than handling and killing only, and repeat surface sterilization methods and barcoding protocol, maybe comparing to other primer sets or barcoding approaches.)  
  
  + Does detection vary over ingestion time based on consumer metabolism? (Approach: variable times of consumer killing post-ingestion.)
  
  + In a broader set of environments, how important is surface contamination? (Approach: surface sterilization of consumers from environments on a gradient of shared surfaces for consumers and prey.)
  
  + What is the optimal surface sterilization approach? (Approach: alter time, concentration, and sterilization method and observe variation in outcome.)
  
#### [Back to the top](#top)

### Supplementary Info

```{r abund PERMANOVA at ID level, include=TRUE}
perm_model1 <- glmmTMB(reads ~ Sterilized + (1+Sterilized|unique_ID), 
                     data = rare_jacc2,
                     family = "genpois")

perm_model_null <- glmmTMB(reads ~ 1 + (1|unique_ID), 
                     data = rare_jacc2,
                     family = "genpois")

AICc(jacc_model1, jacc_model_null)
```

```{r summary perm rare}
summary(perm_model_null)
plot(residuals(perm_model_null))
```

```{r model perm fit}
simulationOutput_perm <- simulateResiduals(fittedModel = perm_model_null)
perm_fit <- plot(simulationOutput_perm, asFactor=TRUE)
```

```{r effects plot of ID perm, include=TRUE}
perm_graph1_u3 <- rare_jacc2 %>%
  filter(pipeline == "u3") %>%
  filter(unique_ID != "Acrididae") %>%
  group_by(unique_ID, Sterilized) %>%
  summarize(mean = mean(reads), sd = sd(reads)) %>%
  ungroup() %>%
  group_by(unique_ID) %>%
  pivot_wider(names_from = c(Sterilized),
              values_from = c(mean, sd)) %>%
  mutate(se_NS = sd_NS/sqrt(19), se_SS = sd_SS/sqrt(18))

es_ID_perm <- esc_mean_se(grp1m = perm_graph1_u3$mean_NS, grp1se = perm_graph1_u3$se_NS, grp1n = 19,
  grp2m = perm_graph1_u3$mean_SS, grp2se = perm_graph1_u3$se_SS, grp2n = 18, es.type = "g")

IDs <- as.character(perm_graph1_u3$unique_ID)
Hedges_g <- es_ID_perm$es
Lower_CI <- es_ID_perm$ci.lo
Upper_CI <- es_ID_perm$ci.hi

effects <- as.data.frame(cbind(IDs, Hedges_g, Lower_CI, Upper_CI))

effects$Hedges_g <- as.numeric(as.character(effects$Hedges_g))
effects$Lower_CI <- as.numeric(as.character(effects$Lower_CI))
effects$Upper_CI <- as.numeric(as.character(effects$Upper_CI))

u3_effect_graph_perm <- ggplot(effects, aes(x = IDs, y = Hedges_g)) +
  geom_point() +theme_bw() +geom_hline(yintercept = 0) +
  geom_pointrange(aes(ymin = Lower_CI, ymax = Upper_CI)) +
  labs(title = "UNOISE3 effect size of average abundance") +
  theme(axis.text.x = element_text(angle=90, hjust = 1)) + coord_flip() 
u3_effect_graph_perm
```


```{r lab rarefied jaccard, include = TRUE}
u3_lab <- u3_comm_rare %>% 
  left_join(u3_id_all, by = "ASV") %>%
  dplyr::select(ASV, all_of(lab), ID_bold, ID_ncbi, taxonomy) %>%
  filter(taxonomy == "prey") #42 prey ASVs
#(first, combine uniques) ##
#use d2_fld and u3_fld with uniques instructions below
#UNOISE3
#make the ID columns characters for the ifelse statements below
u3_lab$ID_bold <- as.character(u3_lab$ID_bold)
u3_lab$ID_ncbi <- as.character(u3_lab$ID_ncbi)

#ifelse statement that assigns each ASV to a specific taxonomy
#I want the BOLD ids to be the last option because these are to
#species in many cases where NCBI/Genbank blasted to order, family, or genus
#therefore, I'm going to say that if there is no BOLD id, give it the NCBI
#ID, otherwise, give it the BOLD ID.
IDs_u3_lab <- ifelse(is.na(u3_lab$ID_bold), u3_lab$ID_ncbi, 
                                               u3_lab$ID_bold)
IDs_u3_lab

#make this a dataframe so we can attach it back to the community
#matrix for analyses at the sample level and later Jaccard dissimiliarity
IDs_u3_lab <- as.data.frame(IDs_u3_lab)
IDs_u3_lab$ASV <- u3_lab$ASV
IDs_u3_lab <- rename(IDs_u3_lab, "unique_ID" = "IDs_u3_lab")
unique_ID_lab <- as.data.frame(unique(IDs_u3_lab[c("unique_ID")]))
unique_ID_lab$sp_number <- seq.int(nrow(unique_ID_lab))

lab_u3 <- IDs_u3_lab %>%
  left_join(unique_ID_lab, by = "unique_ID") %>%
  left_join(u3_lab, by = "ASV") %>%
  gather(sample, reads, HEV07:HEV29) %>%
  left_join(metadata, by = "sample")

rare_jacc_lab <- lab_u3 %>%
  dplyr::select(unique_ID, sample, reads, Sterilized) %>% #select factors of interest
  ungroup() %>% #ungroup any grouping variables
  group_by(sample, unique_ID, Sterilized) %>% #regroup by sample, taxonomiy, and sterilizaion
  summarize(reads = sum(reads)) %>% #summarize the number of reads of each unique taxonomy acros samples
  mutate(pipeline = "u3", presence = ifelse(reads > 0, 1, 0)) #add pipeline so we could combine with each other, and then also create a presence column
 
rare_jac_lab <- rare_jacc_lab %>%
  ungroup() %>%
  mutate(unique_ID = as.factor(unique_ID))

levels(rare_jac_lab$unique_ID)
```

```{r lab remove zeros, include = FALSE}
present_rare_lab <- rare_jac_lab %>%
  group_by(unique_ID) %>%
  tally(reads > 0) %>%
  filter(n > 0) #12 of 28 uniques have non-zero values across all samples in rarefied dataset
```


```{r lab at ID level, include=TRUE}
rare_lab2 <- present_rare_lab %>%
  left_join(rare_jac_lab, by = "unique_ID")
  
jacc_lab1 <- glmmTMB(presence ~ Sterilized + (1+Sterilized|unique_ID), 
                     data = rare_lab2,
                     family = "binomial")

jacc_lab_null <- glmmTMB(presence ~ 1 + (1|unique_ID), 
                     data = rare_lab2,
                     family = "binomial")

AICc(jacc_lab1, jacc_lab_null)
```


```{r summary jaccard rare lab}
summary(jacc_lab_null)
binned_residuals(jacc_lab_null)
```

```{r model jaccard fit lab}
simulationOutput_jacc <- simulateResiduals(fittedModel = jacc_lab_null)
jacc_fit <- plot(simulationOutput_jacc, asFactor=TRUE)
```

```{r effects plot of ID lab, include=TRUE}
jacc_graph2_u3 <- rare_lab2 %>%
  filter(pipeline == "u3") %>%
  filter(unique_ID != "Acrididae") %>%
  group_by(unique_ID, Sterilized) %>%
  summarize(mean = mean(presence), sd = sd(presence)) %>%
  ungroup() %>%
  group_by(unique_ID) %>%
  pivot_wider(names_from = c(Sterilized),
              values_from = c(mean, sd)) %>%
  mutate(se_NS = sd_NS/sqrt(19), se_SS = sd_SS/sqrt(18))

es_ID_u3 <- esc_mean_se(grp1m = jacc_graph2_u3$mean_NS, grp1se = jacc_graph2_u3$se_NS, grp1n = 11,
  grp2m = jacc_graph2_u3$mean_SS, grp2se = jacc_graph2_u3$se_SS, grp2n = 8, es.type = "g")

IDs <- as.character(jacc_graph2_u3$unique_ID)
Hedges_g <- es_ID_u3$es
Lower_CI <- es_ID_u3$ci.lo
Upper_CI <- es_ID_u3$ci.hi

effects <- as.data.frame(cbind(IDs, Hedges_g, Lower_CI, Upper_CI))

effects$Hedges_g <- as.numeric(as.character(effects$Hedges_g))
effects$Lower_CI <- as.numeric(as.character(effects$Lower_CI))
effects$Upper_CI <- as.numeric(as.character(effects$Upper_CI))

u3_effect_graph_lab <- ggplot(effects, aes(x = IDs, y = Hedges_g)) +
  geom_point() +theme_bw() +geom_hline(yintercept = 0) +
  geom_pointrange(aes(ymin = Lower_CI, ymax = Upper_CI)) +
  labs(title = "UNOISE3 effect size of average presence") +
  theme(axis.text.x = element_text(angle=90, hjust = 1)) + coord_flip() 

u3_effect_graph_lab
```

```{r abund PERMANOVA at ID level lab, include=TRUE}
perm_model1 <- glmmTMB(reads ~ Sterilized + (1+Sterilized|unique_ID), 
                     data = rare_lab2,
                     family = "genpois")

perm_model_null <- glmmTMB(reads ~ 1 + (1|unique_ID), 
                     data = rare_lab2,
                     family = "genpois")

AICc(jacc_model1, jacc_model_null)
```

```{r summary perm rare lab}
summary(perm_model_null)
plot(residuals(perm_model_null))
```

```{r model perm fit lab}
simulationOutput_perm <- simulateResiduals(fittedModel = perm_model_null)
perm_fit <- plot(simulationOutput_perm, asFactor=TRUE)
```

```{r effects plot of ID perm lab, include=TRUE}
perm_graph1_u3 <- rare_lab2 %>%
  filter(pipeline == "u3") %>%
  filter(unique_ID != "Acrididae") %>%
  group_by(unique_ID, Sterilized) %>%
  summarize(mean = mean(reads), sd = sd(reads)) %>%
  ungroup() %>%
  group_by(unique_ID) %>%
  pivot_wider(names_from = c(Sterilized),
              values_from = c(mean, sd)) %>%
  mutate(se_NS = sd_NS/sqrt(19), se_SS = sd_SS/sqrt(18))

es_ID_perm <- esc_mean_se(grp1m = perm_graph1_u3$mean_NS, grp1se = perm_graph1_u3$se_NS, grp1n = 19,
  grp2m = perm_graph1_u3$mean_SS, grp2se = perm_graph1_u3$se_SS, grp2n = 18, es.type = "g")

IDs <- as.character(perm_graph1_u3$unique_ID)
Hedges_g <- es_ID_perm$es
Lower_CI <- es_ID_perm$ci.lo
Upper_CI <- es_ID_perm$ci.hi

effects <- as.data.frame(cbind(IDs, Hedges_g, Lower_CI, Upper_CI))

effects$Hedges_g <- as.numeric(as.character(effects$Hedges_g))
effects$Lower_CI <- as.numeric(as.character(effects$Lower_CI))
effects$Upper_CI <- as.numeric(as.character(effects$Upper_CI))

u3_effect_graph_perm_lab <- ggplot(effects, aes(x = IDs, y = Hedges_g)) +
  geom_point() +theme_bw() +geom_hline(yintercept = 0) +
  geom_pointrange(aes(ymin = Lower_CI, ymax = Upper_CI)) +
  labs(title = "UNOISE3 effect size of average abundance") +
  theme(axis.text.x = element_text(angle=90, hjust = 1)) + coord_flip() 
u3_effect_graph_perm_lab
```